<a href="https://colab.research.google.com/github/diwert-ai/Feller-Book-Tasks/blob/main/Feller_book_tasks_(volume_1).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Глава V.<br/>Условная вероятность. Стохастическая независимость.**

## **Задача № 1** 🎲

### <mark>Условие.</mark>

Бросаются три игральные кости. Какова вероятность того, что на одной из них выпадет 1, если на всех трех выпали разные грани.

#### <mark>Решение.</mark>

Дискретное пространство элементарных событий $\mathfrak{S}$ задачи содержит $6^3$ равновероятных событий.
Требуется найти условную вероятность $P(A|H)$, где событие $A$ - на одной из трех костей выпала 1, а $H$ - на всех трех выпали разные грани. Т.к. $P(A|H) = \frac{P(AH)}{P(H)}$, то найдем количество элементарных событий образующих события $AH$ и $H$: $N_{AH} = 6*5*4 = 120$ и $N_{H} = 5*4*3 = 60$. Т.к. все элементарные события равновероятны, то отношение $N_{AH}/N_{H}$ и есть искомая условная вероятность. Таким образом $P(A|H) = N_{AH}/N_{H} = 60/120 = 0.5$

## **Задача № 2** 🎲

### <mark>Условие.</mark>

Известно, что при бросании десяти игральных костей выпала хотя бы одна единица. Какова вероятность $p$ того, что выпало две и более единиц.

#### <mark>Решение.</mark>

Дискретное пространство элементарных событий $\mathfrak{S}$ задачи содержит $6^{10}$ равновероятных событий. Искомая вероятность $p$ является условной вероятностью $P(A|H)=P(AH)/P(H)$, где событие $H$ - "выпало не меньше одной единицы", а событие $A$ - "выпало не меньше двух единиц". <br/>
Найдем $P(H)$. Если $p_0$ - это вероятность того, что единица ни разу не выпала, то $P(H)= 1 - p_0$. Очевидно $p_0 = 5^{10}/6^{10}$ и следовательно $P(H) = \frac{6^{10}-5^{10}}{6^{10}}$<br/>
Найдем $P(AH)$. Заметим, что $AH = A$. И если $p_1$ - это вероятность, что выпадет ровно одна единица, то $P(AH) = P(A) = 1 - p_0 - p_1 = P(H) - p_1$. При этом очевидно $p_1 = {10 \choose 1}\frac{1}{6}\left(\frac{5}{6}\right)^9$.<br/>
Таким образом $p = P(A)/P(H) = 1 - p_1/P(H) = 1 - \frac{10 \cdot5^9}{6^{10} - 5^{10}}$. Вычисление ниже:

In [ ]:
n = 6 ** 10 - 5 ** 10
print(f'p = {1 - 10 * 5 ** 9 / n:.2f}...')

p = 0.61...


## **Задача № 3** 🃏

### <mark>Условие.</mark>

Бридж. При сдаче карт для игры в бридж игрок "Запад" не получил ни одного туза. Какова вероятность того, что его партнер а) не имеет тузов б) имеет не менее двух тузов. Проверить результат не используя формулу условной вероятности.

#### <mark>Решение.</mark>

В игре бридж существует $N = \frac{52!}{(13!)^4}$ разных вариантов раздачи карт четырем игрокам. Считая все варианты равновероятными, имеем дискретное пространство элементарных событий $\mathfrak{S}$, в котором N точек. Вероятность каждой точки $N^{-1}$.<br/><br/>
a) Пусть $H$ - это событие "Игрок "Запад" не получил ни одного туза", а $A$ - "Его партнер (игрок "Восток") не имеет тузов". Тогда искомая вероятность будет $P(A|H) = P(AH)/P(H)$, так как все элементарные события равновероятны, то  $P(A|H) = N_{AH}/N_{H}$ - отношению количеств элементарных событий в $AH$ и $H$. Найдем эти количества.<br/>
$N_H = {48 \choose 13}{39 \choose 13}{26 \choose 13}$ (*Запад получает 13 карт из 48 (колода без тузов), затем следующий игрок получает 13 карт из 39, 3-й игрок получает 13 карт из 26 и 4-му достается оставшиеся 13 карт*)<br/>
$N_{AH} = {48 \choose 13}{35 \choose 13}{26 \choose 13}$ (*Запад получает 13 карт из 48 (колода без тузов), затем Восток получает 13 карт из 35 (остаток колоды без тузов), 3-й игрок получает 13 карт из 26 и 4-му достается оставшиеся 13 карт*)</br>
Таким образом искомая вероятность есть $p_a=N_{AH}/N_{H} = \frac{23\cdot 24\cdot 25\cdot 26}{36\cdot 37\cdot 38\cdot 39}$. Вычисление ниже:

In [ ]:
print(f'p_a = {23*24*25*26/36/37/38/39:.3f}...')

p_a = 0.182...


б) Событие $H$ остается таким же. Событие $A$ теперь - "Игрок "Восток" имеет не менее двух тузов". Найдем новое значение $N_{AH}= {48 \choose 13}\left({4 \choose 2}{35 \choose 11} + {4 \choose 3}{35 \choose 10} + {4 \choose 4}{35 \choose 9}\right){26 \choose 13}$ (*в этом случае у "Востока" есть 3 варианта: 2,3 или 4 туза при раздаче - это дает три слагаемых во втором множителе*).<br/>
Таким образом искомая вероятность равна $p_b=\frac{{4 \choose 2}{35 \choose 11} + {4 \choose 3}{35 \choose 10} + {4 \choose 4}{35 \choose 9}}{{39 \choose 13}}$. Вычисление ниже:

In [ ]:
from scipy.special import comb
a = 6 * comb(35, 11) + 4 * comb(35, 10) + comb(35, 9)
b = comb(39, 13)
print(f'p_b = {a/b:.3f}...')

p_b = 0.407...


С другой стороны, если учесть, что условнaя вероятность того, что "Восток" получит ровно 1 туз будет $p_1 =  \frac{{4 \choose 1}{35 \choose 12}}{{39 \choose 13}}$, а случай 0 тузов уже найден в a), то тот же ответ получим, вычислив $p_b = 1 - p_a - p_1$. Вычисление ниже:

In [ ]:
p_a = 23*24*25*26/36/37/38/39
p_1 = 4 * comb(35, 12) / comb(39, 13)
print(f'p_b = {1 - p_a - p_1:.3f}...')

p_b = 0.407...


## **Задача № 4** 🃏

### <mark>Условие.</mark>

Бридж. Игроки "Север" и "Юг" имеют вместе десять козырей (козыри - карты определенной масти). а) Найти вероятность, что все три остальных козыря находятся у одного игрока (т.е. либо игрок "Восток", либо игрок "Запад" не имеют козырей). б) Известно, что среди трех оставшихся козырей имеется король. Найти вероятность того, что он "не защищен" (т.е. один игрок имеет короля, другой - два остальных козыря).

#### <mark>Решение.</mark>

a) Рассмотрим два события:
* $H = \{$Игроки "Север" и "Юг" имеют вместе десять козырей$\}$
* $A = \{$Либо игрок "Восток", либо "Запад" не имеют козырей$\}$

Тогда требуется найти условную вероятность $P(A|H) = P(AH)/P(H)$. Событие $C=AH$ означает, что либо у игрока "Восток" ровно три козыря (событие $C_e$), либо у игрока "Запад" ровно три козыря (событие $C_w$). Пусть $N_h, N_e, N_w$ количество элементарных событий в $H, C_e, C_w$ соответственно. Считаем, что все элементарные события (варианты раздачи) равновероятны, и т.к. в силу симметрии $N_e = N_w$ (обозначим это кол-во далее $N_c$), то $P(A|H) = \frac{P(C_e\cup C_w)}{P(H)} = \frac{2N_c}{N_h}$. При этом:
* $N_c = {13 \choose 3}{39 \choose 10}{29 \choose 13}$
(*Пояснение формулы: ${13 \choose 3}$ способов выбрать $3$ козыря для первого игрока из $13$ возможных козырей умноженное на ${39 \choose 10}$ способов выбрать оставшиеся $10$ карт для первого игрока из $39$ карт других мастей умноженное на ${29 \choose 13}$ способов выбрать $13$ карт для второго игрока из оставшихся $29$ карт других мастей).*
* $N_h = {13 \choose 3}{39 \choose 23}{26 \choose 13}$
(*Пояснение формулы: заметим, что кол-во способов получить $10$ козырей на двоих для "Севера" и "Юга" равно такому же кол-ву получить $3$ козыря на двоих для "Востока" и "Запада", поэтому $N_h = $ ${13 \choose 3}$ способов выбрать $3$ козыря для двух игроков из $13$ возможных  умноженное на ${39 \choose 23}$ способов выбрать оставшиеся $23$ карты для двух игроков из $39$ карт других мастей и умноженное на ${26 \choose 13}$ способов разделить выбранные $26$ карт по $13$ на каждого из двух игроков).* 

Таким образом $P(A|H) = \frac{2\cdot 11\cdot 12 \cdot 13}{24\cdot 25\cdot 26} = \frac{11}{50}$<br/><br/>
б)  Рассмотрим события:
* $H = \{$У игроков "Восток" и "Запад" на двоих $3$ козыря, один из них король$\}$
* $A = \{$Либо у игрока "Восток", либо у игрока "Запад" только один козырь и это король$\}$


Тогда необходимо найти условную вероятность $P(A|H) = P(AH)/P(H)$. Событие $C = AH$ означает, что либо у игрока "Восток" козырный король, а у игрока "Запад" два других козыря (событие $C_e$), либо наоборот (событие $C_w$). Пусть $N_h, N_e, N_w$ количество элементарных событий в $H, C_e, C_w$ соответственно. Считаем, что все элементарные события (варианты раздачи) равновероятны, и т.к. в силу симметрии $N_e = N_w$ (обозначим это кол-во далее $N_c$), то $P(A|H) = \frac{P(C_e\cup C_w)}{P(H)} = \frac{2N_c}{N_h}$. При этом:
* $N_c = {1 \choose 1}{39 \choose 12}{12 \choose 2}{27 \choose 11}$ *(Пояснение формулы: ${1 \choose 1}$ способ выбрать козырного короля для первого игрока умноженный на ${39 \choose 12}$ способов выбрать оставшиеся $12$ не козырных карт из $39$ для первого игрока умноженное на ${12 \choose 2}$ способов выбрать $2$ козырные карты из $12$ оставшихся для второго игрока умноженное на ${27 \choose 11}$ способов выбрать оставшиеся $11$ карт из $27$ не козырных карт для второго игрока).*
* $N_h = {1 \choose 1}{12 \choose 2}{39 \choose 23}{26 \choose 13}$ *(Пояснение формулы: ${1 \choose 1}$ способ выбрать козырного короля умноженный на ${12 \choose 2}$ способов выбрать еще $2$ козыря из $12$ оставшихся для двух игроков умноженное на ${39 \choose 23}$ способов выбрать оставшиеся $23$ не козырных карты из $39$ для двух игроков умноженное ${26 \choose 13}$ способов разделить выбранные $26$ карт по $13$ на каждого из двух игроков).*

Таким образом $P(A|H) = \frac{2\cdot 12\cdot 13\cdot 13}{24\cdot 25\cdot 26} = \frac{13}{50}$

Ниже проверка вычислений:

In [ ]:
from scipy.special import comb
n_h = comb(39, 23) * comb(26, 13)
n_a = 2 * comb(39, 10) * comb(29, 13)
print(f'а) P(A|H) = {n_a/n_h:.2f}, 2*11*12*13/24/25/26 = 11/50 = {2*11*12*13/24/25/26}')

n_h = comb(39, 23) * comb(26, 13)
n_a = 2 * comb(39, 12) * comb(27, 11)
print(f'б) P(A|H) = {n_a/n_h}, 2*12*13*13/24/25/26 = 13/50 = {2*12*13*13/24/25/26}')

а) P(A|H) = 0.22, 2*11*12*13/24/25/26 = 11/50 = 0.22
б) P(A|H) = 0.26, 2*12*13*13/24/25/26 = 13/50 = 0.26


## **Задача № 6** 🏭🔩

### <mark>Условие.</mark>

На заводе, изготовляющем болты, на долю машин $A$, $B$ и $C$ приходится соответственно 25, 35 и 40% всех изделий. В их продукции брак составляет соответственно 5, 4 и 2%. Случайно выбранный из продукции болт оказался дефектным. Каковы вероятности того, что он был изготовлен на машине $A$? на машине $B$? на машине $C$?

#### <mark>Решение.</mark>

Рассмотрим следующие события:
* $A = \{$болт изготовлен на заводе $A\}$
* $B = \{$болт изготовлен на заводе $B\}$
* $C = \{$болт изготовлен на заводе $C\}$
* $H = \{$болт бракован$\}$

Тогда по условию задачи нам известны следующие вероятности:

* $P(A) = 0.25$
* $P(B) = 0.35$
* $P(C) = 0.40$
* $P(H|A) = 0.05$
* $P(H|B) = 0.04$
* $P(H|C) = 0.02$

Требуется найти условные вероятности:

* $P(A|H)$
* $P(B|H)$
* $P(C|H)$

Найдем первую из них. По определению $P(A|H) = \frac{P(AH)}{P(H)}$. Учитывая, что $P(HA = AH) = P(H|A)P(A)$ получаем $P(A|H) = \frac{P(H|A)P(A)}{P(H)}$. Осталось найти $P(H)$. Т.к. события $A$, $B$ и $C$ делят все пространство событий на непересекающиеся подмножества, то $P(H) = P(H|A)P(A) + P(H|B)P(B) + P(H|C)P(C)$.<br/>
Таким образом $P(A|H) = \frac{P(H|A)P(A)}{P(H|A)P(A) + P(H|B)P(B) + P(H|C)P(C)}$.<br/>
Аналогично находятся две другие вероятности. Вычисление ниже:

In [ ]:
p_a, p_b, p_c = 0.25, 0.35, 0.4
p_ha, p_hb, p_hc =  0.05, 0.04, 0.02
q_a, q_b, q_c = p_a*p_ha, p_b*p_hb, p_c*p_hc 
print(f'P(A|H) = 125/345 = {1/(1+q_b/q_a+q_c/q_a):.4f}...')
print(f'P(B|H) = 140/345 = {1/(1+q_a/q_b+q_c/q_b):.4f}...')
print(f'P(C|H) = 80/345 = {1/(1+q_a/q_c+q_b/q_c):.4f}...')

P(A|H) = 125/345 = 0.3623...
P(B|H) = 140/345 = 0.4058...
P(C|H) = 80/345 = 0.2319...


## **Задача № 7** 👫🏻

### <mark>Условие.</mark>

Предположим, что 5 мужчин из 100 и 25 женщин из 10 000 являются дальтониками. Наугад выбранное лицо страдает дальтонизмом. Какова вероятность того, что это мужчина? (Считать мужчин и женщин одинаковое число).

#### <mark>Решение.</mark>

Обозначим события:
* $D$ - "случайно выбранный человек - дальтоник"
* $M$ - "случайно выбранный человек - мужчина"
* $W$ - "случайно выбранный человек - женщина". 

Тогда из условия:
* $P(M) = P(W) = 0.5$ 
* $P(D|M) = 0.05$
* $P(D|W) = 0.0025$

Требуется найти $P(M|D)$.<br/>
По определению $P(M|D) = P(MD)/P(D) = \frac{P(M)P(D|M)}{P(D)}$.<br/>
Т.к. события $M$ и $W$ делят все пространство элементарных событий на два непересекающихся множества, то $P(D) = P(D|M)P(M) + P(D|W)P(W)$<br/>
Таким образом $P(M|D) = \frac{P(M)P(D|M)}{P(D|M)P(M) + P(D|W)P(W)}$. Вычисление ниже:


In [ ]:
p_m = p_w = 0.5
p_dm, p_dw = 0.05, 0.0025
q_m, q_w = p_m*p_dm, p_w*p_dw
print(f'P(M|D) = 20/21 = {1/(1+q_w/q_m):.4f}...')


P(M|D) = 20/21 = 0.9524...


## **Задача № 11** 👨‍👩‍👧‍👦


### <mark>Условие.</mark>

Пусть<a name="cite_ref-1"></a>[<sup>[1]</sup>](#cite_note-1) вероятность $p_n$ того, что в семье ровно $n$ детей, равна $\alpha p^n$ при $n\ge 1$, и $p_0 = 1 -  \alpha p (1 + p + p^2 + \dots)$. Предположим, что все комбинации полов $n$ детей равновероятны. Показать, что при $k\ge 1$ вероятность того, что в семье ровно $k$ мальчиков, равна ${2\alpha p^k}/{(2-p)^{k+1}}$ 
<br/>
<br/>
<a name="cite_note-1"></a>1. [^](#cite_ref-1) Согласно Лотка, американская семейная статистика удовлетворяет этой гипотезе с $p = 0.7358$. См. Lotka A.J. Théorie analytique des associations biologiques, v. 2, Actualites scientifiques et industrielles, No. 780(1939), Paris, Hermann et Cie. 

#### <mark>Решение.</mark>

1. Учитывая равновероятность комбинаций полов детей при заданном $n$ дискретное пространство элементарных событий $\mathfrak{S}$ будет устроено следующим образом:

| точка пространства <br/>(распределение полов) | вероятность |
| --- | --- |
| 0 | $p_0$ |
| м | $\frac{p_1}{2}$ |
| д | $\frac{p_1}{2}$ |
| мм | $\frac{p_2}{2^2}$ |
| мд | $\frac{p_2}{2^2}$ |
| дм | $\frac{p_2}{2^2}$ |
| дд | $\frac{p_2}{2^2}$ |
| ммм | $\frac{p_3}{2^3}$ |
| ммд | $\frac{p_3}{2^3}$ |
| мдм | $\frac{p_3}{2^3}$ |
| мдд | $\frac{p_3}{2^3}$|
| дмм | $\frac{p_3}{2^3}$|
| дмд | $\frac{p_3}{2^3}$|
| ддм | $\frac{p_3}{2^3}$|
| ддд | $\frac{p_3}{2^3}$|
| и т.д. ...|...|
| ... |... |

<br/><br/>
2. Сначала найдем искомую вероятность $P_k$ для случая $k=1$.
<br/><br/>
Вероятность того, что в семье ровно 1 мальчик будет складываться из суммы вероятностей точек $\mathfrak{S}$, в обозначении которых буква 'м' входит ровно 1 раз. Т.е.:<br/>
$$P_1 = \alpha (\frac{1}{2}p + \frac{2}{2^2}p^2 + \frac{3}{2^3}p^3 + \dots) = \frac{\alpha p}{2}\sum\limits_{i=0}^{\infty}{{i+1 \choose 1}(\frac{p}{2})^i}\space\space\space (*)$$<br/>
где ${i+1 \choose 1}$ биномиальный коэффициент. <br/><br/>
Так как для расширенного биномиального коэффициента справедливо ${-r \choose i} = (-1)^i{r+i-1 \choose i}$, то разложение Тейлора функции $f(x) = (1+x)^{-2}$ имеет вид $f(x) = \sum\limits_{i=0}^{\infty}{{-2 \choose i}x^i} = \sum\limits_{i=0}^{\infty}{{i+1 \choose i}(-1)^i x^i} = \sum\limits_{i=0}^{\infty}{{i+1 \choose 1}(-1)^i x^i}\space\space\space(**)$.<br/><br/>
Заметим также, что $(*)$ можно переписать в виде:<br/>
$$P_1 = \frac{\alpha p}{2}\sum\limits_{i=0}^{\infty}{{i+1 \choose 1}(-1)^i(-\frac{p}{2})^i}$$<br/><br/>Учитывая $(**)$ получаем:<br/>
$$P_1 = \frac{\alpha p}{2}(1 - \frac{p}{2})^{-2} = 2\alpha p/(2 - p)^2$$
<br/><br/>
3. По аналогии рассмотрим теперь $P_2$. Вероятность того, что в семье ровно 2 мальчика будет складываться из суммы вероятностей точек $\mathfrak{S}$, в обозначении которых буква 'м' встречается дважды. Т.е.:<br/>
$$P_2 = \alpha\left({2 \choose 2}\left(\frac{p}{2}\right)^2 + {3 \choose 2}\left(\frac{p}{2}\right)^3 + \dots \right) = \frac{\alpha p^2}{4}\sum\limits_{i=0}^{\infty}{{i+2 \choose 2}\left(\frac{p}{2}\right)^i} = \frac{\alpha p^2}{4}\sum\limits_{i=0}^{\infty}{{i+2 \choose 2}(-1)^i\left(-\frac{p}{2}\right)^i}$$<br/><br/>
С другой стороны:<br/>
$$(1+x)^{-3} = \sum\limits_{i=0}^{\infty}{{-3 \choose i}x^i} = \sum\limits_{i=0}^{\infty}{{i+2 \choose 2}(-1)^i x^i}$$<br/>
Поэтому:<br/>
$$P_2 = \frac{\alpha p^2}{4}\left(1 - \frac{p}{2}\right)^{-3} = 2\alpha p^2/(2 - p)^3$$<br/><br/>
4. И наконец рассмотрим общий случай $P_k$:<br/>
$$P_k = \alpha\left({k \choose k}\left(\frac{p}{2}\right)^k + {k+1 \choose k}\left(\frac{p}{2}\right)^{k+1} + \dots \right) = \frac{\alpha p^k}{2^k}\sum\limits_{i=0}^{\infty}{{i+k \choose k}(-1)^i\left(-\frac{p}{2}\right)^i}$$<br/>
И т.к.:<br/>
$$(1+x)^{-(k+1)} = \sum\limits_{i=0}^{\infty}{{i+k \choose k}(-1)^i x^i}$$<br/>
то:<br/>
$$P_k = \frac{\alpha p^k}{2^k}\left(1 - \frac{p}{2}\right)^{-(k+1)} = 2\alpha p^k/(2 - p)^{k+1}$$ что и требовалось показать.

# **Глава VI.<br/>Биномиальное распределение и распределение Пуассона.**

## **Задача № 1** 👨‍👩‍👧‍👦

### <mark>Условие.</mark>

Предполагая все комбинации полов детей равновероятными, определить, какую долю семей с 
шестью детьми будут составлять семьи с тремя мальчиками и тремя девочками.

#### <mark>Решение.</mark>

Будем считать, что рождение очередного ребенка в семье - это испытание Бернулли в серии из $n = 6$ испытаний. Пускай успех - это рождение мальчика. Т.к. все комбинации полов детей равновероятны, то вероятность успеха $p = 0.5$. Требуется найти вероятность $k = 3$ успехов, т.е. $b(3; 6, 0.5) = {6 \choose 3}\cdot 0.5^3\cdot 0.5^3$. Вычисление ниже:

In [ ]:
print(f'5/16 = {20*(0.5)**6}')

5/16 = 0.3125


## **Задача № 2** 🃏

### <mark>Условие.</mark>

Игрок в бридж при трех последовательных сдачах не получил туза. Есть ли у него основания жаловаться на невезение.

#### <mark>Решение.</mark>

Будем считать каждую сдачу испытанием Бернулли в серии из $n = 3$ испытаний. Пускай успехом будет отсутствие туза в сдаче, а $p$ вероятность успеха, тогда вероятность события, которое произошло с игроком - это $k = 3$ успеха в серии из $n = 3$ испытаний, т.е. $b(3;3,p) = p^3$. Найдем $p$.<br/>
Всего различных вариантов сдач для игрока будет $N = {52 \choose 13}$ (*кол-во способов выбрать 13 карт из 52*), а вариантов сдач без туза будет $N_0 = {48 \choose 13}$ (*кол-во способов выбрать 13 карт из 48 (колода без тузов)*). Т.к. все сдачи равновероятны, то $p = N_0/N = \frac{36\cdot 37\cdot 38\cdot 39}{49\cdot 50\cdot 51\cdot 52}$. Таким образом вероятность события, которое случилось с игроком будет: $\left(\frac{36\cdot 37\cdot 38\cdot 39}{49\cdot 50\cdot 51\cdot 52}\right)^3$. Вычисление ниже:

In [ ]:
p = (36*37*38*39/49/50/51/52) ** 3
print(f'p = {p:.5f}...')

p = 0.02804...


Как можно видеть событие весьма редкое, случается в среднем примерно в 3 сдачах из 100.

## **Задача № 3** 🔢

### <mark>Условие.</mark>

Сколько случайных цифр нужно взять, чтобы вероятность появления среди них цифры $7$ была не меньше $9/10$.

#### <mark>Решение.</mark>

Будем считать ряд случайных цифр серией $n$ испытаний Бернулли. Считая появление любой цифры из $10$ возможных равновероятным, а появление цифры $7$ успехом, получим, что вероятность успеха будет $p = 0.1$. Если $p_0$ - это вероятность $k = 0$ успехов, то искомая вероятность $P_7 = 1 - p_0$. При этом $p_0 = b(0; n, p) = (1 - p)^n $. Т.к. $P_7 \ge 9/10$, то получаем:<br/>
$1 - (1 - p)^n \ge 9/10 \Rightarrow n \ge \frac{\ln 0.1}{\ln 0.9}$. Вычисление ниже:


In [ ]:
from math import log, ceil
print(f'n >= {ceil(log(0.1)/log(0.9))}')

n >= 22


## **Задача № 4** 🃏

### <mark>Условие.</mark>

Сколько раз нужно сдать колоду карт для игр в бридж (сдачи независимы), чтобы вероятность того, что фиксированный игрок по меньшей мере один раз получит четыре туза, была не меньше $1/2$. Решить задачу и в том случае, когда игрок заранее не фиксируется.

#### <mark>Решение.</mark>

Будем рассматривать серию сдач как серию $n$ испытаний Бернулли. Успехом будем считать, то что фиксированный игрок на сдаче получил 4 туза. Считая, что любая сдача равновероятна, вероятность успеха будет $p = \frac{{4 \choose 4}{48 \choose 9}}{{52 \choose 13}} = \frac{10\cdot 11\cdot 12\cdot 13}{49\cdot 50\cdot 51\cdot 52}$ (*в числителе кол-во успешных сдач: число способов выбрать 4 туза из 4-х умноженное на кол-во способов выбрать остальные 9 карт из 48-ми (колода без тузов), а в знаменателе все возможные сдачи: число способов выбрать 13 карт из 52-х*).<br/>
Если $p_0$ - это вероятность получить $k = 0$ успехов в серии из $n$ испытаний, то $p_0 = b(0; n, p) = (1 - p)^n$. При этом $1 - p_0$ - это вероятность того, что игрок хотя бы раз получит 4 туза на сдаче. Таким образом, получаем:<br/>
$1 - p_0 \ge 0.5 \Rightarrow n \ge \frac{\ln 0.5}{\ln(1-p)}$, где $p = \frac{10\cdot 11\cdot 12\cdot 13}{49\cdot 50\cdot 51\cdot 52}$.<br/>
В случае, когда игрок не фиксирован вероятность успеха будет в 4 раза больше (по числу игроков), т.е. $4p$ и поэтому искомое кол-во испытаний будет $n_4 \ge \frac{\ln 0.5}{\ln(1-4*p)}$. Вычисление ниже:

In [ ]:
from math import log, ceil
p = 10*11*12*13/49/50/51/52
print(f'Игрок фиксирован: n >= {ceil(log(0.5)/log(1-p))}')
print(f'Игрок не фиксирован: n_4 >= {ceil(log(0.5)/log(1-4 * p))}')

Игрок фиксирован: n >= 263
Игрок не фиксирован: n_4 >= 66


## **Задача № 5** 🎯

### <mark>Условие.</mark>

Пусть вероятность попадания в цель равна $1/5$ и производится $10$ независимых выстрелов. Какова вероятность попадания в цель по меньшей мере дважды?

#### <mark>Решение.</mark>

Будем считать серию из $10$ выстрелов испытаниями Бернулли. Вероятность успеха $p = 0.2$. А искомая вероятность $p_2 = 1 - b(0; 10, p) - b(1; 10, p) = 1 - (0.8)^{10} - {10 \choose 1}\cdot 0.2\cdot (0.8)^9$. Вычисление ниже:

In [ ]:
print(f'p_2 = {1 - (0.8) ** 10 - 2*(0.8) ** 9:.4f}...')

p_2 = 0.6242...


## **Задача № 6** 🎯

### <mark>Условие.</mark>

Найти в задаче $5$ условную вероятность попадания в цель по меньшей мере дважды, если известно, что по крайней мере одно попадание произошло.

#### <mark>Решение.</mark>

Рассмотрим следующие события:
* $A$ - "произошло больше одного попадания"
* $H$ - "произошло по меньшей мере одно попадание"

Тогда искомая вероятность будет $P(A|H) = \frac{P(AH)}{P(H)}$. При этом $P(AH) = P(A) = p_2$, где $p_2$ найдено в задаче $5$. А $P(H) = 1 - b(0; 10, 0.2)$. Таким образом $P(A|H) = \frac{1 - (0.8)^{10} - {10 \choose 1}\cdot 0.2\cdot (0.8)^9}{1 - (0.8)^{10}}$. Вычисление ниже:

In [ ]:
p_ah = 1 - (0.8) ** 10 - 2*(0.8) ** 9
p_h = 1 - (0.8) ** 10 
print(f'P(A|H) = {p_ah/p_h:.4f}...')

P(A|H) = 0.6993...


## **Задача № 7** 🃏

### <mark>Условие.</mark>

Найти вероятность того, что среди 13 карт, наугад выбранных из колоды в 52 карты, имеются ровно 2 карты красной масти. Сравнить ее с соответствующей вероятностью для испытаний Бернулли с $p = 1/2$.

#### <mark>Решение.</mark>

Всего элементарных событий ровно столько сколькими способами можно выбрать 13 карт из 52-х, т.е. ${52 \choose 13}$. Событий, вероятность которых необходимо найти, будет равно кол-ву способов выбрать 2 красных карты из 26 возможных красных помноженное на кол-во способов выбрать оставшиеся 11 черных карты из 26 возможных черных или ${26 \choose 2}{26 \choose 11}$. Считая все элементарные события равновероятными получаем, что искомая вероятность: $\frac{{26 \choose 2}{26 \choose 11}}{52 \choose 13}$. Вычисление ниже:

In [ ]:
from math import factorial
def comb(n, k):
  return factorial(n) // factorial(n-k) // factorial(k)

print(f'= {comb(26, 2) * comb(26, 11) / comb(52, 13):.6f}...')

= 0.003954...


Если допустить, что очередной выбор карты из колоды является испытанием Бернулли с вероятностью успеха $p=1/2$, то искомая вероятность будет равна $b(2;13, 1/2)$. Вычисление ниже.

In [ ]:
from scipy.stats import binom
print(f'= {binom.pmf(2, 13, 0.5):.5f}...')

= 0.00952...


Как видно, второй ответ существенно отличается от первого. Ведь в данном случае выбор карты из колоды испытанием Бернулли не является, т.к. вероятность успеха очередного выбора зависит от результатов предыдущих испытаний.

## **Задача № 8** 🎂

### <mark>Условие.</mark>

Какова вероятность того, что дни рождения шести людей приходятся на два месяца. Предполагается независимость и равновероятность всех месяцев.

#### <mark>Решение.</mark>

Допустим два месяца выбраны заранее. Будем считать месяца рождения шести людей равновероятными результатами шести испытаний Бернулли. Успехом будем считать исход испытания, когда месяц рождения человека приходится на один из двух заранее выбранных месяцев. Вероятность такого успеха для одного испытания будет $p_1 = 1/6$. При этом совпадение месяца рождения у всех шести людей сразу с одним из двух заранее выбранных месяцев будем считать шестью "ложными успехами". Вероятность одного такого "ложного успеха" будет $p_2 = 1/12$. Таким образом для заранее выбранной пары месяцев искомая вероятность будет $p_0 = b(6; 6, p_1) - 2*b(6; 6, p_2)$. Так как всего существует ${12 \choose 2}$ разных способов выбрать 2 месяца из 12, то искомая вероятность будет $p = {12 \choose 2}p_0 = 66 * (1/6^6 - 2/12^6)$. Вычисление ниже:

In [ ]:
p_1 = 1 / 6
p_2 = 1 / 12
print(f'p = {66 * (p_1 ** 6 - 2 * p_2 ** 6):.5f}...')

p = 0.00137...


#### <mark>Замечание.</mark>

Можно было бы рассуждать иначе. Будем считать серией из $12$ испытаний следующую процедуру.
* **Первое** испытание: фиксируем неуспех в случае, если месяц рождения **январь** либо у всех шести человек, либо ни у кого. В противном случае фиксируем успех.
* **Второе** испытание: фиксируем неуспех в случае, если месяц рождения **февраль** либо у всех шести человек, либо ни у кого. В противном случае фиксируем успех.

И так далее для всех 12 месяцев. Тогда вероятность успеха в первом испытании будет $p = 1 - (11/12)^6 - (1/12)^6$. И поэтому можно было бы решить, что вероятность ровно двух успехов в серии из 12 испытаний будет $b(2; 12, p) = {12 \choose 2}p^2(1-p)^{10}$. Однако вычисление ниже показывает вероятность на порядок большую, чем та что получена ранее:

In [ ]:
p = 1 - (11/12) ** 6 - (1/12) ** 6
print(f'вероятность успеха в первом испытании: {p:.4f}...\nвероятность ровно 2х успехов из 12: {66 * (p ** 2) * ((1 - p)**10):.4f}...')

вероятность успеха в первом испытании: 0.4067...
вероятность ровно 2х успехов из 12: 0.0590...


Все дело в том, что рассмотренная процедура не является схемой Бернулли. Потому что группа из шести человек зафиксирована заранее и следовательно вероятность успеха очередного испытания зависит от результатов предыдущих испытаний, что не допустимо.

## **Задача № 9** 🎲

### <mark>Условие.</mark>

Найти вероятность того, что при бросании $6$ правильных костей единица выпадает а) по меньшей мере один раз, б) ровно один раз, в) ровно два раза. Сравнить эти вероятности с их пуассоновскими приближениями. 

#### <mark>Решение.</mark>

Будем рассматривать бросание костей как серию $n = 6$ испытаний Бернулли. Исходы равновероятны и следовательно вероятность успеха $p = 1/6$. Тогда искомые вероятности и их пуассоновские приближения при $\lambda = np = 1$ будут следующие:
* a) $1 - b(0; 6; 1/6);\space\space\space\space 1 - p(0; 1)$
* б) $b(1; 6; 1/6);\space\space\space\space p(1; 1)$
* в) $b(2; 6; 1/6);\space\space\space\space p(2; 1)$

Вычисление ниже:

In [ ]:
from scipy.stats import binom, poisson
n, p, lam = 6, 1/6, 1
print(f'a) истинное значение: {1 - binom.pmf(0, n, p):.4f}...; пуассоновское приближение: {1 - poisson.pmf(0, lam):.4f}...;')
print(f'б) истинное значение: {binom.pmf(1, n, p):.4f}... ; пуассоновское приближение: {poisson.pmf(1, lam):.4f}... ;')
print(f'в) истинное значение: {binom.pmf(2, n, p):.4f}... ; пуассоновское приближение: {poisson.pmf(2, lam):.4f}... .')

a) истинное значение: 0.6651...; пуассоновское приближение: 0.6321...;
б) истинное значение: 0.4019... ; пуассоновское приближение: 0.3679... ;
в) истинное значение: 0.2009... ; пуассоновское приближение: 0.1839... .


## **Задача № 10** 🧑‍🤝‍🧑

### <mark>Условие.</mark>

Известно, что левши составляют в среднем $1\%$. Оценить вероятность того, что среди $200$ людей окажется по меньшей мере четверо левшей.

#### <mark>Решение.</mark>

Рассмотрим серию $n = 200$ испытаний Бернулли с вероятностью успеха $p = 0.01$. Требуется найти вероятность числа успехов $k \ge 4$. Искомая вероятность будет $p_b = 1 - b(0; 200, p) - b(1; 200, p) - b(2; 200, p) - b(3; 200, p)$. Пуассоновское приближение при $\lambda = np = 2$ будет $p_p = 1 - p(0; 2) - p(1; 2) - p(2; 2) - p(3; 2)$. Вычисление ниже:

In [ ]:
from scipy.stats import binom, poisson
n, p = 200, 0.01
lam = n * p
pb = pp = 1
for k in range(4):
  pb -= binom.pmf(k, n, p)
  pp -= poisson.pmf(k, lam)
print(f'Бернуллиевское значение: {pb:.3f}...\nПуассоновское приближение: {pp:.3f}...')

Бернуллиевское значение: 0.142...
Пуассоновское приближение: 0.143...


## **Задача № 11** 📖

### <mark>Условие.</mark>

Книга в 500 страниц содержит 500 опечаток. Оценить вероятность того, что на заданной странице не менее трех опечаток.

#### <mark>Решение.</mark>

Наличие или отсутствие опечаток на странице книги можно считать пространственным распределением случайных событий в объеме $t$ (в нашем случае $t$ - это объем одной страницы). При этом допустимо считать, что:<br/>
1) при малых размерах $t$ вероятность иметь более одной опечатки мала по сравнению с $t$<br/>
2) неперекрывающиеся области страницы взаимно независимы.<br/>
Поэтому вероятность допустимо оценить с помощью распределения Пуассона $p(k; \lambda t)$.<br/><br/>
Оценим $\lambda t$. Имеется $N = 500$ независимых экспериментов (по количеству страниц в книге). Общее число событий (опечаток) наблюдаемых во всех $N$ экспериментах $T = 500$. Поэтому $\lambda t \approx T/N = 1$. Таким образом искомая оценка вероятности будет равна $1 - p(0; 1) - p(1; 1) - p(2; 1)$. Вычисление ниже:

In [ ]:
from scipy.stats import poisson
lam, ps = 1, 1
for k in range(3):
    ps -= poisson.pmf(k, lam)
print(f' = {ps:.3f}...')

 = 0.080...


## **Задача № 12** 🧑‍🤝‍🧑

### <mark>Условие.</mark>

Дальтоники составляют $1\%$ лиц в некоторой группе людей. Как велика должна быть случайная выборка (с возвращением), чтобы вероятность иметь в ней одного дальтоника была не меньше $0.95$

#### <mark>Решение.</mark>

Рассмотрим серию $n$ испытаний Бернулли с вероятностью успеха $p = 0.01$. По условию $1 - b(0; n_b, p) \ge 0.95$, и аналогично для пуассоновского приближения $1 - p(0; n_pp) \ge 0.95$. Отсюда получаем:
* $n_b \ge \frac{\ln 0.05}{\ln 0.99}$
* $n_p \ge \frac{\ln 0.05}{-0.01}$

In [ ]:
from math import log, ceil
print(f'Бернуллиевское значение n_b >= {ceil(log(0.05)/log(0.99))}')
print(f'Пуассоновское приближение n_p >= {ceil(log(0.05)/-0.01)}')

Бернуллиевское значение n_b >= 299
Пуассоновское приближение n_p >= 300


## **Задача № 13** 🧑‍🤝‍🧑

### <mark>Условие.</mark>

В предыдущей задаче найти вероятность того, что в выборке из $100$ человек a) нет дальтоников, б) дальтоников два или больше. 

#### <mark>Решение.</mark>

Рассмотрим серию $n = 100$ испытаний Бернулли с вероятностью успеха $p = 0.01$. Тогда:
* a) бернуллиевское значение $p^a_b = b(0; 100, p)$ пуассовновское приближение $p^a_p = p(0, 100p)$
* б) бернуллиевское значение $p^b_b = 1 - b(0; 100, p) - b(1; 100, p)$ пуассовновское приближение $p^b_p = 1 - p(0, 100p) - p(1; 100p)$

Вычисление ниже:

In [ ]:
from scipy.stats import binom, poisson
n, p = 100, 0.01
lam = n * p
b_0 = binom.pmf(0, 100, p)
b_1 = binom.pmf(1, 100, p)
p_0 = poisson.pmf(0, lam)
p_1 = poisson.pmf(1, lam)
print(f'a) бурнуллиевское: {b_0:.4f}...; пуассоновское: {p_0:.4f}...')
print(f'б) бурнуллиевское: {1 - b_0 - b_1:.4f}...; пуассоновское: {1 - p_0 - p_1:.4f}...')

a) бурнуллиевское: 0.3660...; пуассоновское: 0.3679...
б) бурнуллиевское: 0.2642...; пуассоновское: 0.2642...


## **Задача № 14** 🍪

### <mark>Условие.</mark>

Оценить среднее число изюминок, которое должно быть в одной булочке, чтобы не более одной булочки из ста было без изюма.

#### <mark>Решение.</mark>

Наличие или отсутствие изюминок в булочке можно считать пространственным распределением случайных событий в объеме $t$ (в нашем случае  $t$ - это объем одной булочки). При этом допустимо считать, что:<br/>
1) при малых размерах  $t$  вероятность иметь более одной изюминки мала по сравнению с  $t$<br/>
2) неперекрывающиеся области булочки взаимно независимы.<br/>
Поэтому оценку кол-ва изюминок можно сделать с помощью распределения Пуассона  $p(k;λt)$. Параметр $\lambda t$ оценивается как $T/N$, где $T$ это общее количество изюминок в $N$ булочках, т.е. $\lambda t \approx$ искомое среднее число изюминок. Обозначим его $\lambda_t$. Тогда $p(k;\lambda_t)$ есть вероятность, что в случайно выбранной булочке будет ровно $k$ изюминок. Так как требуется чтобы булочка без изюминок встречалась не чаще, чем 1 раз на 100, то $p(0, \lambda_t)\le 0.01$, т.е. $e^{-\lambda_t}\le 0.01$. Таким образом $\lambda_t \ge -\ln{0.01}$. Вычисление ниже:

In [ ]:
from math import log, ceil
print(f'>= {ceil(-log(0.01))}')

>= 5


Т.е. среднее число изюминок можно оценить как $\ge 5$

# **Глава IX.<br/>Случайные величины. Математическое ожидание.**

## **Пример г). Время ожидания при выборе.** ⏲️
*цитата из книги*

Предварительное замечание (*из Примера в*):<br/>
Пусть $\textbf{X}$ - случайная величина, имеющая *геометрическое распределение* $\textbf{P}\{\textbf{X}=k\} = q^kp$, где $k=0, 1, 2, \dots $. Тогда $\textbf{E}(\textbf{X}) = qp(1+2q+3q^2+\dots)$. Справа в скобках стоит производная геометрического ряда, поэтому $\textbf{E}(\textbf{X})=qp(1-q)^{-2}=q/p$. $\textbf{X}$ можно трактовать как число неудач до первого успеха в последовательности испытаний Бернулли.

А теперь сам пример:<br/>
Из генеральной совокупности, содержащей $N$ различных элементов, производится выбор с возвращением. Вследствие возможных повторений случайная выборка объема $r$ в общем случае будет состоять из менее чем $r$ различных элементов. При увеличении объема выборки новые элементы будут появляться в ней все реже и реже. Нас интересует, каков должен быть объем выборки $\textbf{S}_r$, чтобы в ней оказалось $r$ различных элементов.<br/>
Для упрощения изложения назовем очередное извлечение элемента успешным, если выбранный элемент появляется в выборке впервые. Тогда $\textbf{S}_r$ представляет собой число извлечений до $r$-го успеха. Положим $\textbf{X}_k = \textbf{S}_{k+1} - \textbf{S}_k$. Тогда $\textbf{X}_k-1$ есть число неудачных извлечений между $k$-м и $(k+1)$-м успехами. Во время этих извлечений в генеральной совокупности было $N-k$ элементов, которые еще не появлялись в выборке, и поэтому $X_k-1$ равно числу неудач до первого успеха в испытаниях Бернулли с $p=(N-k)/N$. Следовательно, согласно замечанию выше, имеем $\textbf{E}(\textbf{X}_k)=1 + q/p=N/(N-k)$. Наконец, поскольку $\textbf{S}_r=1+\textbf{X}_1+\dots+\textbf{X}_r$ получаем $$\textbf{E}(\textbf{S}_r)=N\left(1/N+1/(N-1)+\dots+1/(N-r+1)\right)$$
В частности, $\textbf{E}(\textbf{S}_N)$ есть ожидаемое число извлечений, необходимых для того, чтобы в выборку вошли все элементы генеральной совокупности. При $N=10$ имеем $\textbf{E}(\textbf{S}_5)\approx 6,5$ и $\textbf{E}(\textbf{S}_{10})\approx 29,3$. Вычисления ниже:



In [ ]:
def expectation_when_choosing(gen_n, r):
  sum = 1/gen_n
  for i in range(1, r):
    sum += 1/(gen_n-i)
  return gen_n * sum

print(f'E(S(10, 5)) ~ {expectation_when_choosing(10, 5):.1f}')
print(f'E(S(10, 10)) ~ {expectation_when_choosing(10, 10):.1f}')

E(S(10, 5)) ~ 6.5
E(S(10, 10)) ~ 29.3


Это означает, что для выборки первой половины совокупности из $10$ элементов в среднем достаточно семи извлечений, а на вторую половину потребуется еще $23$ извлечения.<br/>
Чтобы получить приближенную формулу, будем трактовать $(N-k)^{-1}$ как площадь прямоугольника, основанием которого является интервал единичной длины с центром в точке $N-k$, а высотой - значение функции $x^{-1}$ в этой точке. Заменяя площадь этого прямоугольника на площадь области, лежащей ниже графика функции $x^{-1}$, получаем приближение
$$\textbf{E}(\textbf{S}_r)\approx N \int\limits_{N-r+1/2}^{N+1/2}x^{-1}\,dx = N\log{[(N+1/2)/(N-r+1/2)]}$$
В качестве примера использования этого приближения возьмем произвольное $\alpha \lt 1$ и рассмотрим *ожидаемое число извлечений, необходимых для получения в выборке доли $\alpha$ всей генеральной совокупности*. Оно равно $\textbf{E}(\textbf{S}_r)$, где $r$ - наименьшее целое $\ge \alpha N$. При $N \to \infty$ погрешность, допускаемая при использовании приближения, стремится к нулю и искомое математическое ожидание в пределе равно $N\log(1-\alpha)^{-1}$.

## **Пример д). Задача об оценке.** 🧐
*цитата из книги*

Некоторая урна содержит шары с номерами от $1$ до $N$. Пусть $\textbf{X}$ - *наибольший номер, полученный при* $n$ извлечениях, если производится случайный выбор с возвращением. Событие $\textbf{X} \le k$ означает, что каждый из $n$ вынутых номеров не превышает $k$, и, следовательно, $\textbf{P}\{\textbf{X} \le k\}=(k/N)^n$. Таким образом, распределение вероятностей случайной величины $\textbf{X}$ определяется формулой
$$p_k=\textbf{P}\{\textbf{X}=k\} = \textbf{P}\{\textbf{X} \le k\} - \textbf{P}\{\textbf{X} \le k-1\} = \left(k^n-(k-1)^n\right)N^{-n}.$$
Отсюда следует, что
$$\textbf{E}(\textbf{X}) = \sum\limits_{k=1}^{N}{kp_k}=N^{-n}\left(\sum\limits_{k=1}^{N}{k^{n+1}-(k-1)^{n+1}-(k-1)^{n}}\right) = N^{-n}\left(N^{n+1}-\sum\limits_{k=1}^{N}{(k-1)^n}\right)$$
Если в городе имеется $N=1000$ машин и производится случайная выборка $n=10$, то математическое ожидание максимального наблюденного регистрационного номера приблеженно равно $910$. Вычисление ниже:

In [ ]:
from math import ceil

def highest_expected_number(gen_n, n):
  sum = 0
  for i in range(1, gen_n):
    sum += (i-1) ** n
  
  return (gen_n ** (n+1) - sum)/(gen_n ** n)

print(f'N={1000} n={10} -> hen={int(highest_expected_number(1000, 10))}')

N=1000 n=10 -> hen=910


При больших $N$ последняя сумма приближенно равна площади области, лежащей ниже кривой $y=x^n$ и ограниченной прямыми $x=0$ и $x=N$, т.е. $N^{n+1}/(n+1)$. Отсюда следует, что при больших $N$
$$\textbf{E}(\textbf{X})\approx [n/(n+1)]N$$

In [ ]:
def appr_hen(gen_n, n):
  return n*gen_n/(n+1)

print(f'N={1000} n={10} -> hen={round(appr_hen(1000, 10))}')

N=1000 n=10 -> hen=909


<mark>В прикладной статистике наблюденный максимум выборки используется для оценки неизвестного истинного значения $N$. Этот метод применялся во время второй мировой войны для оценки производства противника.</mark> 

## **Задача № 1** 🎱

### <mark>Условие.</mark>

Семь шаров случайно распределяются по семи ящикам. Пусть $\textbf{X}_i$ - число ящиков, содержащих ровно $i$ шаров. Используя вероятности из таблицы $\S 5$ гл. II, выписать совместное распределение $\textbf{X}_2$ и $\textbf{X}_3$.

#### <mark>Решение.</mark>

Согласно таблицы $\S 5$ гл. II получаем совместное распределение:<br/>
$ P\{\textbf{X}_2=0,\ \textbf{X}_3=0 \} = 0.006120 + 0.035699 + 0.005355 + 0.000357 + 0.000008 = 0.047539 $<br/>
$ P\{\textbf{X}_2=1,\ \textbf{X}_3=0 \} = 0.128518 + 0.026775 + 0.001071 = 0.156364 $<br/>
$ P\{\textbf{X}_2=2,\ \textbf{X}_3=0 \} = 0.321295 $<br/>
$ P\{\textbf{X}_2=3,\ \textbf{X}_3=0 \} = 0.107098 $<br/>
$ P\{\textbf{X}_2=0,\ \textbf{X}_3=1 \} = 0.108883 $<br/>
$ P\{\textbf{X}_2=1,\ \textbf{X}_3=1 \} = 0.214197 $<br/>
$ P\{\textbf{X}_2=2,\ \textbf{X}_3=1 \} = 0.026775 $<br/>
$ P\{\textbf{X}_2=3,\ \textbf{X}_3=1 \} = 0 $<br/>
$ P\{\textbf{X}_2=0,\ \textbf{X}_3=2 \} = 0.017850 $<br/>
$ P\{\textbf{X}_2=1,\ \textbf{X}_3=2 \} = 0 $<br/>
$ P\{\textbf{X}_2=2,\ \textbf{X}_3=2 \} = 0 $<br/>
$ P\{\textbf{X}_2=3,\ \textbf{X}_3=2 \} = 0 $<br/><br/>
И маргинальные (частные) распределения:<br/>
$ P\{\textbf{X}_3=0\} = 0.047539 + 0.156364 + 0.321295 + 0.107098 = 0.632296 $<br/>
$ P\{\textbf{X}_3=1\} = 0.108883 + 0.214197 + 0.026775 = 0.349855 $<br/>
$ P\{\textbf{X}_3=2\} = 0.017850 $<br/><br/>
$ P\{\textbf{X}_2=0\} = 0.047539 + 0.108883 + 0.017850 = 0.174272 $<br/>
$ P\{\textbf{X}_2=1\} = 0.156364 + 0.214197 = 0.370561 $<br/>
$ P\{\textbf{X}_2=2\} = 0.321295 + 0.026775 = 0.34807 $<br/>
$ P\{\textbf{X}_2=3\} = 0.107098 $<br/>

Вычисления ниже:

In [ ]:
print(f'P(x2=0, x3=0) ~ {0.006120 + 0.035699 + 0.005355 + 0.000357 + 0.000008:.6}')
print(f'P(x2=1, x3=0) ~ {0.128518 + 0.026775 + 0.001071:.6}')
print(f'P(x2=0, x3=1) ~ {0.107098 + 0.001785:.6}')
print(f'P(x3=0) ~ {0.047539 + 0.156364 + 0.321295 + 0.107098:.6}')
print(f'P(x3=1) ~ {0.108883 + 0.214197 + 0.026775:.6}')
print(f'P(x3=0,1,2) ~ {0.632296 + 0.349855 + 0.017850:.6}')
print(f'P(x2=0) ~ {0.047539 + 0.108883 + 0.017850:.6}')
print(f'P(x2=1) ~ {0.156364 + 0.214197:.6}')
print(f'P(x2=2) ~ {0.321295 + 0.026775:.6}')
print(f'P(x2=0,1,2,3) ~ {0.174272 + 0.370561 + 0.34807 + 0.107098:.6}')

P(x2=0, x3=0) ~ 0.047539
P(x2=1, x3=0) ~ 0.156364
P(x2=0, x3=1) ~ 0.108883
P(x3=0) ~ 0.632296
P(x3=1) ~ 0.349855
P(x3=0,1,2) ~ 1.0
P(x2=0) ~ 0.174272
P(x2=1) ~ 0.370561
P(x2=2) ~ 0.34807
P(x2=0,1,2,3) ~ 1.0


## **Задача № 2** 🎲

### <mark>Условие.</mark>

Бросаются две правильные кости. Пусть $\textbf{X}$ - число очков на первой кости и $\textbf{Y}$ - бо́льшее из двух выпавших чисел. а) Выписать совместное распределение $\textbf{X}$ и $\textbf{Y}$. б) Найти математические ожидания, дисперсии и ковариацию.

#### <mark>Решение.</mark>

Легко заметить, что а)

$$\begin{matrix}
    \textbf{P}\{\textbf{X}=j,\ \textbf{Y}=k\} & =
    & \left\{
    \begin{matrix}
    0 & \mbox{если } j \gt k \\
    j/36 & \mbox{если } j = k \\
    1/36 & \mbox{если } j \lt k \\
    \end{matrix} \right.
\end{matrix}$$
Поэтому б)<br/> 
$ \textbf{E}(\textbf{X}) = 7/2 $<br/>
$ \textbf{E}(\textbf{X}^2) = 91/6 $<br/>
$ \mathrm{Var}(\textbf{X}) = 35/12 $<br/>
$ \textbf{E}(\textbf{Y}) = 161/36 $<br/>
$ \textbf{E}(\textbf{Y}^2) = 791/36 $<br/>
$ \mathrm{Var}(\textbf{Y}) = 2555/1296 $<br/>
$ \textbf{E}(\textbf{XY}) = 154/9 $<br/>
$ \mathrm{Cov}(\textbf{X}, \textbf{Y}) = 105/72 = 35/24 $<br/>

Вычисления ниже:

In [ ]:
import numpy as np
from fractions import Fraction

values = np.array(range(1, 7))          # возможные значения сл. величин X и Y
n = len(values)                         # кол-во разных значений   
p_xy = np.zeros((n, n), dtype=object)   # таблица вероятностей совместного распределения X и Y
p_x = np.zeros(n, dtype=object)         # таблица маргинального распределения X
p_y = np.zeros(n, dtype=object)         # таблица маргинального распределения Y

# заполняем таблицы всех распределений
for j in values:
  p_x[j - 1] = Fraction(1, n)
  p_y[j - 1] = Fraction(2 * j - 1, n ** 2)
  for k in values[j - 1:]:
    p_xy[j - 1, k - 1] = Fraction(j, n * n) if j == k else Fraction(1, n * n)

# проверка корректности (все суммы должны быть 1)
print(f'check sums: {p_xy.sum()} {p_x.sum()} {p_y.sum()}')

e_x = np.dot(values, p_x)                              # мат. ожидание X
e_y = np.dot(values, p_y)                              # мат. ожидание Y
e_xx = np.dot(values ** 2, p_x)                        # момент X 2-го порядка
e_yy = np.dot(values ** 2, p_y)                        # момент Y 2-го порядка
v_x = e_xx - e_x ** 2                                  # дисперсия X
v_y = e_yy - e_y ** 2                                  # дисперсия Y
xy_values = [i * j for i in values for j in values]    # список возможных значений XY
e_xy = np.dot(xy_values, p_xy.flatten())               # мат. ожидание XY
c_xy = e_xy - e_x * e_y                                # ковариация X, Y

print(f'E(X) = {e_x}\nE(X^2) = {e_xx}\nE(Y) = {e_y}\nE(Y^2) = {e_yy}')
print(f'Var(X) = {v_x}\nVar(Y) = {v_y}')
print(f'E(XY) = {e_xy}\nCov(X, Y) = {c_xy}')

check sums: 1 1 1
E(X) = 7/2
E(X^2) = 91/6
E(Y) = 161/36
E(Y^2) = 791/36
Var(X) = 35/12
Var(Y) = 2555/1296
E(XY) = 154/9
Cov(X, Y) = 35/24


## **Задача № 3** 🪙

### <mark>Условие.</mark>

Пусть $\textbf{X},\ \textbf{Y},\ \textbf{Z}$ - соответственно число выпадений герба, число серий гербов и длина максимальной из этих серий при пяти бросаниях монеты. Составить таблицу из $32$ элементарных событий с соответствующищи значениями $\textbf{X}$, $\textbf{Y}$ и $\textbf{Z}$.<br/>
Простым подсчетом найти совместное распределение пар  $(\textbf{X},\ \textbf{Y}),\ (\textbf{X},\ \textbf{Z}),\ (\textbf{Y},\ \textbf{Z})$. Вычислить математические ожидания, дисперсии и ковариации этих случайных величин. 

#### <mark>Решение.</mark>

Пространство элементарных событий  $\mathfrak{S}$ содержит $32$ точки. Точка - это последовательность нулей и единиц длиной $5$, где $0$ означает выпадение герба, a $1$ - решки. Таким образом (с помощью кода, который приведен ниже) получаем следующую таблицу значений  $\textbf{X},\ \textbf{Y},\ \textbf{Z}$ на точках  $\mathfrak{S}$.

| номер<br/>эл. события | эл. событие | $\textbf{X}$ | $\textbf{Y}$ | $\textbf{Z}$ | 
| --- | --- | --- | --- | --- |
| 00 | 00000 | 5 | 1 | 5 |
| 01 | 00001 | 4 | 1 | 4 |
| 02 | 00010 | 4 | 2 | 3 |
| 03 | 00011 | 3 | 1 | 3 |
| 04 | 00100 | 4 | 2 | 2 |
| 05 | 00101 | 3 | 2 | 2 |
| 06 | 00110 | 3 | 2 | 2 |
| 07 | 00111 | 2 | 1 | 2 |
| 08 | 01000 | 4 | 2 | 3 |
| 09 | 01001 | 3 | 2 | 2 |
| 10 | 01010 | 3 | 3 | 1 |
| 11 | 01011 | 2 | 2 | 1 |
| 12 | 01100 | 3 | 2 | 2 |
| 13 | 01101 | 2 | 2 | 1 |
| 14 | 01110 | 2 | 2 | 1 |
| 15 | 01111 | 1 | 1 | 1 |
| 16 | 10000 | 4 | 1 | 4 |
| 17 | 10001 | 3 | 1 | 3 |
| 18 | 10010 | 3 | 2 | 2 |
| 19 | 10011 | 2 | 1 | 2 |
| 20 | 10100 | 3 | 2 | 2 |
| 21 | 10101 | 2 | 2 | 1 |
| 22 | 10110 | 2 | 2 | 1 |
| 23 | 10111 | 1 | 1 | 1 |
| 24 | 11000 | 3 | 1 | 3 |
| 25 | 11001 | 2 | 1 | 2 |
| 26 | 11010 | 2 | 2 | 1 |
| 27 | 11011 | 1 | 1 | 1 |
| 28 | 11100 | 2 | 1 | 2 |
| 29 | 11101 | 1 | 1 | 1 |
| 30 | 11110 | 1 | 1 | 1 |
| 31 | 11111 | 0 | 0 | 0 |

Все точки считаем равновероятными, поэтому вероятность каждой точки $p = 1/32$.

Таблицы (код их генерации см. ниже) совместных распределений будут выглядеть следующим образом. (В ячейках с вероятностями подразумевается множитель $p$):

| $\textbf{Y}$ \ $\textbf{X}$ | 0 | 1 | 2 | 3 | 4 | 5 | Распр. $\textbf{Y}$|
| --- | --- | --- | --- | --- | --- | --- | --- |
|0|1|0|0|0|0|0|1|
|1|0|5|4|3|2|1|15|
|2|0|0|6|6|3|0|15|
|3|0|0|0|1|0|0|1|
|Распр. $\textbf{X}$|1|5|10|10|5|1| |
<br/>

| $\textbf{Z}$ \ $\textbf{X}$ | 0 | 1 | 2 | 3 | 4 | 5 | Распр. $\textbf{Z}$|
| --- | --- | --- | --- | --- | --- | --- | --- |
|0|1|0|0|0|0|0|1|
|1|0|5|6|1|0|0|12|
|2|0|0|4|6|1|0|11|
|3|0|0|0|3|2|0|5|
|4|0|0|0|0|2|0|2|
|5|0|0|0|0|0|1|1|
|Распр. $\textbf{X}$|1|5|10|10|5|1| |
<br/>

| $\textbf{Z}$ \ $\textbf{Y}$ | 0 | 1 | 2 | 3 | Распр. $\textbf{Z}$|
| --- | --- | --- | --- | --- | --- |
|0|1|0|0|0|1|
|1|0|5|6|1|12|
|2|0|4|7|0|11|
|3|0|3|2|0|5|
|4|0|2|0|0|2|
|5|0|1|0|0|1|
|Распр $\textbf{Y}$|1|15|15|1| |
<br/>

Таблицы распределений сл. величин $\textbf{X}+\textbf{Y}$ и $\textbf{X}\textbf{Y}$ будут следующие (опять же с точностью до множителя $p$):

|$\textbf{X}+\textbf{Y}$|0|2|3|4|5|6|
| --- | --- | --- | --- | --- | --- | --- |
|$P/p$|1|5|4|9|8|5|
<br/>

|$\textbf{X}\textbf{Y}$|0|1|2|3|4|5|6|8|9|
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
|$P/p$|1|5|4|3|8|1|6|3|1|
<br/>

Мат. ожидания, дисперсии и ковариации сл. величин будут следующие:<br/>
$\textbf{E}(\textbf{X}) = 5/2$<br/>
$\textbf{E}(\textbf{X}^2) = 15/2$<br/>
$\textbf{E}(\textbf{Y}) = 3/2$<br/>
$\textbf{E}(\textbf{Y}^2) = 21/8$<br/>
$\textbf{E}(\textbf{Z}) = 31/16$<br/>
$\textbf{E}(\textbf{Z}^2) = 79/16$<br/>
$\mathrm{Var}(\textbf{X}) = 5/4$<br/>
$\mathrm{Var}(\textbf{Y}) = 3/8$<br/>
$\mathrm{Var}(\textbf{Z}) = 303/256$<br/>
$\textbf{E}(\textbf{X}\textbf{Y}) = 4$<br/>
$\mathrm{Cov}(\textbf{X}, \textbf{Y}) = 1/4$<br/>
$\textbf{E}(\textbf{X}\textbf{Z}) = 47/8$<br/>
$\mathrm{Cov}(\textbf{X}, \textbf{Z}) = 33/32$<br/>
$\textbf{E}(\textbf{Y}\textbf{Z}) = 45/16$<br/>
$\mathrm{Cov}(\textbf{Y}, \textbf{Z}) = -3/32$<br/>
$\textbf{E}(\textbf{X}+\textbf{Y}) = 4$<br/>
$\textbf{E}(\textbf{X}\textbf{Y}) = 4$<br/>
$\mathrm{Var}(\textbf{X}+\textbf{Y}) = 17/8$<br/>
$\mathrm{Var}(\textbf{X}\textbf{Y}) = 89/16$

Вычисления ниже:

In [ ]:
import re
import pandas as pd
import numpy as np
from fractions import Fraction

# функция возвращает значение сл. величин X, Y, Z элементарного события event.
def get_xyz(event: str):
  series = [len(seq) for seq in re.findall('0+', event)]
  return event.count('0'), len(series), max(series) if series else 0

xyz_event_list = [] 

# генерация таблицы значений X, Y, Z на точках пространства эл. событий
print('X,Y,Z')
for i in range(32):
  event = f'{i:05b}'
  x, y, z = get_xyz(event)
  xyz_event_list.append((event, x, y, z))
  print(f'| {i:02} | {event} | {x} | {y} | {z} |')

df_xyz = pd.DataFrame(xyz_event_list, columns=['event', 'X', 'Y', 'Z'])    # создаем фрейм со значениями функций X, Y, Z
x_vals = df_xyz.X.unique()   # возможные значения X
y_vals = df_xyz.Y.unique()   # возможные значения Y
z_vals = df_xyz.Z.unique()   # возможные значения Z

nx, ny, nz = len(x_vals), len(y_vals), len(z_vals)  # кол-ва разных значений X, Y, Z соответственно

x_vals.sort()
y_vals.sort()
z_vals.sort()

p = Fraction(1, 32)   # вероятность эл. события

p_xy = np.zeros((ny, nx), dtype=object)  # таблица вероятностей совместного распределения X и Y
p_xz = np.zeros((nz, nx), dtype=object)  # таблица вероятностей совместного распределения X и Z
p_yz = np.zeros((nz, ny), dtype=object)  # таблица вероятностей совместного распределения Y и Z

# заполняем таблицы совместных распределений
for x in range(nx):
  for y in range(ny):
    p_xy[y, x] = p * len(df_xyz.query(f'Y=={y} and X=={x}'))
  for z in range(nz):
    p_xz[z, x] = p * len(df_xyz.query(f'X=={x} and Z=={z}'))

for y in range(ny):
  for z in range(nz):
    p_yz[z, y] = p * len(df_xyz.query(f'Y=={y} and Z=={z}'))

# вычисляем маргинальные распределения X, Y, Z
p_x = np.sum(p_xy, axis=0)
p_y = np.sum(p_xy, axis=1)
p_z = np.sum(p_xz, axis=1)

# проверка корректности (все суммы должны быть 1)
print()
print('check sums: ',p_xy.sum(), p_xz.sum(), p_yz.sum(), p_x.sum(), p_y.sum(), p_z.sum())
print()

# генерация таблиц совместных распределений (с точностью до множителя p)
def print_joint_distribution(f_vals, s_vals, p_fs, p_f, p_s, f_marker):
  for s in s_vals:
    print(f'|{s}', end='|')
    for f in f_vals:
      print(f'{p_fs[s, f]/p}', end='|')
    print(f'{p_s[s]/p}', end='|')
    print()
  print(f'|{f_marker}:|', end='')
  for f in f_vals:
    print(f'{p_f[f]/p}', end='|')
  print(' |')

print('\nX,Y')
print_joint_distribution(x_vals, y_vals, p_xy, p_x, p_y, 'X')
print('\nX,Z')
print_joint_distribution(x_vals, z_vals, p_xz, p_x, p_z, 'X')
print('\nY,Z')
print_joint_distribution(y_vals, z_vals, p_yz, p_y, p_z, 'Y')

# создаем колонки во фрейме для величин X+Y и XY
df_xyz['X+Y'] = df_xyz.X + df_xyz.Y
df_xyz['XY'] = df_xyz.X * df_xyz.Y

xplusy_vals = df_xyz['X+Y'].unique()  # возможные значения X+Y
xplusy_vals.sort()
xmuly_vals = df_xyz['XY'].unique() # возможные значения XY
xmuly_vals.sort()

n_xplusy, n_xmuly = len(xplusy_vals), len(xmuly_vals) # кол-во разных значений X+Y и XY соответственно

p_xplusy = np.zeros(n_xplusy, dtype = object)  # таблица распределения X+Y
p_xmuly = np.zeros(n_xmuly, dtype = object)    # таблица распределения XY

# заполняем таблицы распределения X+Y и XY
for i, val in enumerate(xplusy_vals):
  p_xplusy[i] = p * len(df_xyz.query(f'X+Y=={val}'))

for i, val in enumerate(xmuly_vals):
  p_xmuly[i] = p * len(df_xyz.query(f'XY=={val}'))

# генерация таблиц распределения X+Y и XY (с точностью до множителя p)
def print_distribution(vals, p_func):
  print('|', end='')
  print(*vals, sep='|', end='')
  print('|')

  print('|', end='')
  print(*p_func/p, sep='|', end='')
  print('|')

print('\nX+Y')
print_distribution(xplusy_vals, p_xplusy)
print('\nXY')
print_distribution(xmuly_vals, p_xmuly)


e_x = np.dot(x_vals, p_x)                              # мат. ожидание X
e_y = np.dot(y_vals, p_y)                              # мат. ожидание Y
e_z = np.dot(z_vals, p_z)                              # мат. ожидание Z
e_xx = np.dot(x_vals ** 2, p_x)                        # момент X 2-го порядка
e_yy = np.dot(y_vals ** 2, p_y)                        # момент Y 2-го порядка
e_zz = np.dot(z_vals ** 2, p_z)                        # момент Z 2-го порядка
v_x = e_xx - e_x ** 2                                  # дисперсия X
v_y = e_yy - e_y ** 2                                  # дисперсия Y
v_z = e_zz - e_z ** 2                                  # дисперсия Z
xy_vals = [x * y for y in y_vals for x in x_vals]      # список возможных значений XY
xz_vals = [x * z for z in z_vals for x in x_vals]      # список возможных значений XZ
yz_vals = [y * z for z in z_vals for y in y_vals]      # список возможных значений YZ
e_xy = np.dot(xy_vals, p_xy.flatten())                 # мат. ожидание XY
e_xz = np.dot(xz_vals, p_xz.flatten())                 # мат. ожидание XZ
e_yz = np.dot(yz_vals, p_yz.flatten())                 # мат. ожидание YZ      
c_xy = e_xy - e_x * e_y                                # ковариация X, Y
c_xz = e_xz - e_x * e_z                                # ковариация X, Z
c_yz = e_yz - e_y * e_z                                # ковариация Y, Z
e_xplusy = np.dot(xplusy_vals, p_xplusy)               # мат. ожидание X+Y
e_xmuly = np.dot(xmuly_vals, p_xmuly)                  # мат. ожидание XY 
e_xplusy2 = np.dot(xplusy_vals ** 2, p_xplusy)         # момент X+Y 2-го порядка 
e_xmuly2 = np.dot(xmuly_vals ** 2, p_xmuly)            # момент XY 2-го порядка
v_xplusy = e_xplusy2 - e_xplusy ** 2                   # дисперсия X+Y
v_xmuly = e_xmuly2 - e_xmuly ** 2                      # дисперсия XY


print(f'\nE(X) = {e_x}\nE(X^2) = {e_xx}\nE(Y) = {e_y}\nE(Y^2) = {e_yy}\nE(Z) = {e_z}\nE(Z^2) = {e_zz}')
print(f'Var(X) = {v_x}\nVar(Y) = {v_y}\nVar(Z) = {v_z}')
print(f'E(XY) = {e_xy}\nCov(X, Y) = {c_xy}')
print(f'E(XZ) = {e_xz}\nCov(X, Z) = {c_xz}')
print(f'E(YZ) = {e_yz}\nCov(Y, Z) = {c_yz}')
print(f'E(X+Y) = {e_xplusy}\nE(XY) = {e_xmuly}')
print(f'Var(X+Y) = {v_xplusy}\nVar(XY) = {v_xmuly}')

X,Y,Z
| 00 | 00000 | 5 | 1 | 5 |
| 01 | 00001 | 4 | 1 | 4 |
| 02 | 00010 | 4 | 2 | 3 |
| 03 | 00011 | 3 | 1 | 3 |
| 04 | 00100 | 4 | 2 | 2 |
| 05 | 00101 | 3 | 2 | 2 |
| 06 | 00110 | 3 | 2 | 2 |
| 07 | 00111 | 2 | 1 | 2 |
| 08 | 01000 | 4 | 2 | 3 |
| 09 | 01001 | 3 | 2 | 2 |
| 10 | 01010 | 3 | 3 | 1 |
| 11 | 01011 | 2 | 2 | 1 |
| 12 | 01100 | 3 | 2 | 2 |
| 13 | 01101 | 2 | 2 | 1 |
| 14 | 01110 | 2 | 2 | 1 |
| 15 | 01111 | 1 | 1 | 1 |
| 16 | 10000 | 4 | 1 | 4 |
| 17 | 10001 | 3 | 1 | 3 |
| 18 | 10010 | 3 | 2 | 2 |
| 19 | 10011 | 2 | 1 | 2 |
| 20 | 10100 | 3 | 2 | 2 |
| 21 | 10101 | 2 | 2 | 1 |
| 22 | 10110 | 2 | 2 | 1 |
| 23 | 10111 | 1 | 1 | 1 |
| 24 | 11000 | 3 | 1 | 3 |
| 25 | 11001 | 2 | 1 | 2 |
| 26 | 11010 | 2 | 2 | 1 |
| 27 | 11011 | 1 | 1 | 1 |
| 28 | 11100 | 2 | 1 | 2 |
| 29 | 11101 | 1 | 1 | 1 |
| 30 | 11110 | 1 | 1 | 1 |
| 31 | 11111 | 0 | 0 | 0 |

check sums:  1 1 1 1 1 1


X,Y
|0|1|0|0|0|0|0|1|
|1|0|5|4|3|2|1|15|
|2|0|0|6|6|3|0|15|
|3|0|0|0|1|0|0|1|
|X:|1|5|10|10|5|1| |

X,

## Задача № 4 🔷

### <mark>Условие.</mark>

Пусть $\textbf{X}$, $\textbf{Y}$ и $\textbf{Z}$ - независимые случайные величины, имеющие одно и то же геометрическое распределение $\{q^kp\}$. Найти 
- а) $\textbf{P}\{\textbf{X}=\textbf{Y}\}$; 
- б) $\textbf{P}\{\textbf{X} \ge 2\textbf{Y}\}$; 
- в) $\textbf{P}\{\textbf{X} + \textbf{Y} \le \textbf{Z}\}$.

#### <mark>Решение.</mark>

- a) Очевидно $p_a=\textbf{P}\{\textbf{X}=\textbf{Y}\} = \sum\limits_{k=0}^{\infty}\textbf{P}\{\textbf{X}=k,\ \textbf{Y}=k\}$ И т.к. $\textbf{X}$, $\textbf{Y}$ независимые сл. величины и $p = 1 - q$, то $p_a = \sum\limits_{k=0}^{\infty}\textbf{P}\{\textbf{X}=k\} \textbf{P}\{\textbf{Y}=k\} = \sum\limits_{k=0}^{\infty}q^kp\cdot q^kp = p^2\sum\limits_{k=0}^{\infty}q^{2k} = p^2 /(1-q^2) = p/(1+q)$
- б) Пользуясь независимостью $\textbf{X}$ и $\textbf{Y}$, получаем $p_b = \textbf{P}\{\textbf{X} \ge 2\textbf{Y}\} = \sum\limits_{k=0}^{\infty}\textbf{P}\left\{\{\textbf{Y}=k\},\ \bigcup\limits_{i=0}^{\infty}\{\textbf{X}=2k+i\}\right\} = \sum\limits_{k=0}^{\infty}q^kp\left(\sum\limits_{i=0}^{\infty}q^{2k+i}p\right) = p^2\sum\limits_{k=0}^{\infty}q^{3k}\left(\sum\limits_{i=0}^{\infty}q^i\right)=p^2\sum\limits_{k=0}^{\infty}q^{3k}/(1-q)=$<br>$=p\sum\limits_{k=0}^{\infty}(q^3)^k = p/(1-q^3) = 1/(1+q+q^2)$
- в) Пользуясь независимостью $\textbf{X}$, $\textbf{Y}$ и $\textbf{Z}$, получаем $p_c = \textbf{P}\{\textbf{X} + \textbf{Y} \le \textbf{Z}\} = \sum\limits_{j,k=0}^{\infty}\textbf{P}\left\{\textbf{X}=j,\ \textbf{Y}=k,\ \bigcup\limits_{i=0}^{\infty}\{\textbf{Z}=k+j+i\}\right\} = \sum\limits_{j,k=0}^{\infty}q^jp\ q^kp\ \left(\sum\limits_{i=0}^{\infty}q^{j+k+i}p\right) = p^3\sum\limits_{j,k=0}^{\infty}q^{2(j+k)}\left(\sum\limits_{i=0}^{\infty}q^i\right) = p^2\sum\limits_{j,k=0}^{\infty}q^{2(j+k)}$<br>Пусть $t = q^2$, тогда последняя сумма будет равна $S_t=1+2t+3t^2+\dots$ ,  которая является производной по $t$ от суммы $t+t^2+t^3+\dots = t/(1-t)$, т.е. $S_t = 1/(1-t)^2$.<br>Таким образом $p_c = p^2/(1-q^2)^2 = 1/(1+q)^2$

## Задача № 5 🔷

### <mark>Условие.</mark>

*Продолжение 4-й задачи.* Пусть $\textbf{U}$ наименьшая из величин $\textbf{X}$ и $\textbf{Y}$. Положим $\textbf{V} = \textbf{X} - \textbf{Y}$. Показать, что $\textbf{U}$ и $\textbf{V}$ независимы$^1$.
<br><br><br>
$^1)$ Геометрическое распределение является единственным определенным на целых числах распределением вероятностей, для которого это утверждение верно. См. Ferguson T.S., A characterization of the geometric distribution, Amer. Math. Monthly, 72 (1965), 256-260. 
https://www.math.ucla.edu/~tom/papers/Geom.pdf

#### <mark>Решение.</mark>

Требуется показать, что $\mathbf{P}\{\mathbf{U}=j,\ \mathbf{V}=k\} = \mathbf{P}\{\mathbf{U}=j\}\mathbf{P}\{\mathbf{V}=k\}$ для любых $j,k \in \{0,1,2\dots\}$. Вычислим левую и правую часть по отдельности.<br>$\mathbf{P}\{\mathbf{U}=j,\ \mathbf{V}=k\} = \mathbf{P}\{\min(\mathbf{X}, \mathbf{Y})=j,\ \mathbf{X}-\mathbf{Y}=k\} = \mathbf{P}\{\mathbf{X}=j+k,\ \mathbf{Y}=j\} = \mathbf{P}\{\mathbf{X}=j+k\}\mathbf{P}\{\mathbf{Y}=j\}=q^{j+k}p\ q^jp = q^{2j+k}p^2$<br><br>
С другой стороны:<br><br>$\mathbf{P}\{\mathbf{U}=j\} = \mathbf{P}\{\min(\mathbf{X}, \mathbf{Y})=j\} = \mathbf{P}\{\mathbf{X}=j,\ \mathbf{Y} \ge j\} + \mathbf{P}\{\mathbf{Y}=j,\ \mathbf{X} \gt j\} = \mathbf{P}\{\mathbf{X}=j\}\mathbf{P}\{\mathbf{Y} \ge j\} + \mathbf{P}\{\mathbf{Y}=j\}\mathbf{P}\{\mathbf{X} \gt j\}=$<br>$= q^jp\ q^jp\sum\limits_{i=0}^{\infty}q^i + q^jp\ q^jp\sum\limits_{i=1}^{\infty}q^i = q^{2i}p^2/(1-q) + q^{2i}p^2q/(1-q)=q^{2j}p(1+q)$<br><br><br>
$\mathbf{P}\{\mathbf{V}=k\} = \mathbf{P}\{\mathbf{X}-\mathbf{Y}=k\}=\sum\limits_{i=0}^{\infty}\mathbf{P}\{\mathbf{X}=i,\ \mathbf{Y}=i+k\}=\sum\limits_{i=0}^{\infty}\mathbf{P}\{\mathbf{X}=i\}\mathbf{P}\{\mathbf{Y}=i+k\} = \sum\limits_{i=0}^{\infty}q^ip\ q^{i+k}p=q^kp^2\sum\limits_{i=0}^{\infty}(q^2)^i=q^kp^2/(1-q^2)=q^kp/(1+q)$
<br><br>
Таким образом<br>
$\mathbf{P}\{\mathbf{U}=j\}\mathbf{P}\{\mathbf{V}=k\} = q^{2j+k}p^2 = \mathbf{P}\{\mathbf{U}=j,\ \mathbf{V}=k\}$, что и требовалось показать.



## Задача № 6 🔷

### <mark>Условие.</mark>

Пусть $\textbf{X}_1$ и $\textbf{X}_2$ - независимые случайные величины, имеющие распределения Пуассона $\{p(k;\lambda_1)\}$ и $\{p(k;\lambda_2)\}$.
- а) Доказать, что $\textbf{X}_1 + \textbf{X}_2$ имеет рапсределение Пуассона $\{p(k;\lambda_1+\lambda_2)\}$
- б) Доказать, что условное распределение $\textbf{X}_1$ при заданной сумме $\textbf{X}_1 + \textbf{X}_2$ есть биномиальное распределение, а именно что $$\textbf{P}\{\textbf{X}_1=k\ |\ \textbf{X}_1 + \textbf{X}_2=n\} = b(k;\ n,\ \lambda_1/(\lambda_1+\lambda_2)).$$

#### <mark>Решение.</mark>

- а) Найдем распределение  вероятностей $\textbf{X}_1 + \textbf{X}_2$, учитывая независимость $\textbf{X}_1$ и $\textbf{X}_2$ :<br>
$\textbf{P}\{\textbf{X}_1 + \textbf{X}_2=n\} =\sum\limits_{k=0}^n\textbf{P}\{\textbf{X}_1=k,\ \textbf{X}_2=n-k\} = \sum\limits_{k=0}^n\textbf{P}\{\textbf{X}_1=k\}\textbf{P}\{\textbf{X}_2=n-k\}=\sum\limits_{k=0}^ne^{-\lambda_1}\frac{\lambda_1^k}{k!}\ e^{-\lambda_2}\frac{\lambda_2^{n-k}}{(n-k)!} = e^{-(\lambda_1+\lambda_2)}\cdot\lambda_2^n\sum\limits_{k=0}^n\left(\frac{\lambda_1}{\lambda_2}\right)^k\frac{1}{k!(n-k)!}$<br>
Если последнюю сумму (обозначим ее $S_n$) домножить на $n!$, то под знаком суммы появятся биномиальные коэффициенты $n \choose k$, т.е. получим биномиальное разложение:
$\left(1+\frac{\lambda_1}{\lambda_2}\right)^n=n!S_n$. Поэтому $$\textbf{P}\{\textbf{X}_1 + \textbf{X}_2=n\} =  \frac{e^{-(\lambda_1+\lambda_2)}}{n!}\cdot\lambda_2^n\left(1+\frac{\lambda_1}{\lambda_2}\right)^n = e^{-(\lambda_1+\lambda_2)}\frac{(\lambda_1+\lambda_2)^n}{n!} = p(n;\ \lambda_1+\lambda_2)$$
Что и требовалось показать.<br>
- б) По определению условной вероятности: $\textbf{P}\{\textbf{X}_1=k\ |\ \textbf{X}_1 + \textbf{X}_2=n\} = \frac{\textbf{P}\{\textbf{X}_1=k,\ \textbf{X}_1 + \textbf{X}_2=n\}}{\textbf{P}\{\textbf{X}_1 + \textbf{X}_2=n\}} = \frac{\textbf{P}\{\textbf{X}_1=k,\ \textbf{X}_2=n-k\}}{\textbf{P}\{\textbf{X}_1 + \textbf{X}_2=n\}} = \frac{e^{-(\lambda_1+\lambda_2)}\lambda_1^k\lambda_2^{n-k}}{e^{-(\lambda_1+\lambda_2)}(\lambda_1+\lambda_2)^n}\frac{n!}{k!(n-k)!} =$<br>$= {n \choose k}\frac{\lambda_1^k\lambda_2^{n-k}}{(\lambda_1+\lambda_2)^k(\lambda_1+\lambda_2)^{n-k}}={n \choose k}\left(\frac{\lambda_1}{\lambda_1+\lambda_2}\right)^k\left(\frac{\lambda_2}{\lambda_1+\lambda_2}\right)^{n-k} = b(k;\ n,\ \lambda_1/(\lambda_1+\lambda_2)).$<br><br>
Что и требовалось показать.

## Задача № 7 🔷

### <mark>Условие.</mark>

Пусть $\textbf{X}_1$ и $\textbf{X}_2$ независимы и имеют одно и то же геометрическое распределение $\{q^kp\}$. Не вычисляя, показать, что *условное распределение $\textbf{X}_1$ при заданной сумме $\textbf{X}_1+\textbf{X}_2$ равномерно*, т.е. что $$\textbf{P}\{\textbf{X}_1=k\ |\ \textbf{X}_1+\textbf{X}_2=n\} = 1/(n+1)$$

#### <mark>Решение.</mark>

$\textbf{P}\{\textbf{X}_1=k\ |\ \textbf{X}_1+\textbf{X}_2=n\} = \frac{\textbf{P}\{\textbf{X}_1=k,\ \textbf{X}_1+\textbf{X}_2=n\}}{\textbf{P}\{\textbf{X}_1+\textbf{X}_2=n\}} = \frac{\textbf{P}\{\textbf{X}_1=k\}\textbf{P}\{\textbf{X}_2=n-k\}}{\sum\limits_{i=0}^{n}\textbf{P}\{\textbf{X}_1=i\}\textbf{P}\{\textbf{X}_2=n-i\}}=\frac{q^kpq^{n-k}p}{q^np^2\sum\limits_{i=0}^n1} =  \frac{1}{n+1}$

## Задача № 8 🔷

### <mark>Условие.</mark>

Пусть $\textbf{X}_1, \textbf{X}_2, \dots, \textbf{X}_r$ - взаимно независимые случайные величины, каждая из которых имеет *равномерное распределение* $\textbf{P}\{\textbf{X}_i=k\}=1/N$ для $k=1, 2, \dots N$. Пусть $\textbf{U}_n$ наименьшая из величин $\textbf{X}_1, \textbf{X}_2, \dots, \textbf{X}_n$, а  $\textbf{V}_n$ - наибольшая из них. Найти распределения $\textbf{U}_n$ и $\textbf{V}_n$. Как связана эта задача с задачей об оценке (пример д))?

#### <mark>Решение.</mark>

Величина $\textbf{V}_n$ распределена также как величина $\textbf{X}$ из примера д). (см. выше - примеры тек. раздела). Действительно, пользуясь взаимной независимостью величин $\textbf{X}_i$  получим, что $\textbf{P}\{\textbf{V}_n \le k\} = \textbf{P}\{\textbf{X}_1 \le k,\ \textbf{X}_2 \le k,\ \dots,\ \textbf{X}_n \le k\} = \textbf{P}\{\textbf{X}_1 \le k\}\textbf{P}\{\textbf{X}_2 \le k\}\dots\textbf{P}\{\textbf{X}_n \le k\} = \prod\limits_{j=1}^n\left(\sum\limits_{i=1}^k\textbf{P}\{\textbf{X}_j=i\}\right) = \left(\frac{k}{N}\right)^n$<br>
С другой стороны $\textbf{P}\{\textbf{V}_n \le k\} = \sum\limits_{i=1}^k\textbf{P}\{\textbf{V}_n = i\}$, поэтому<br>
$\textbf{P}\{\textbf{V}_n = k\} =  \textbf{P}\{\textbf{V}_n \le k\} - \sum\limits_{i=1}^{k-1}\textbf{P}\{\textbf{V}_n = i\} = \textbf{P}\{\textbf{V}_n \le k\} - \textbf{P}\{\textbf{V}_n \le k - 1\} = \frac{k^n-(k-1)^n}{N^n}$<br><br>
Аналогично найдем распределение $\textbf{U}_n$:<br>
$\textbf{P}\{\textbf{U}_n \ge k\} = \textbf{P}\{\textbf{X}_1 \ge k,\ \textbf{X}_2 \ge k,\ \dots,\ \textbf{X}_n \ge k\} =  \textbf{P}\{\textbf{X}_1 \ge k\}\textbf{P}\{\textbf{X}_2 \ge k\}\dots\textbf{P}\{\textbf{X}_n \ge k\} = \prod\limits_{j=1}^n\left(\sum\limits_{i=0}^{N-k}\textbf{P}\{\textbf{X}_j=k+i\}\right) =$<br>$= \left(\frac{N-k+1}{N}\right)^n \Rightarrow \textbf{P}\{\textbf{U}_n = k\} = \textbf{P}\{\textbf{U}_n \ge k\} - \textbf{P}\{\textbf{U}_n \ge k + 1\} = \frac{(N-k+1)^n-(N-k)^n}{N^n}$

## Задача № 9 🔷

### <mark>Условие.</mark>

*Дополнение к задаче об оценке* (пример д)).
- a) Найти совместное распределение максимального и минимального из наблюденных значений. Особо разобрать случай $n=2$. (*Указание.* Сначала вычислить $\textbf{P}\{\textbf{X} \le r,\ \textbf{Y} \ge s\}$.)
- б) Найти условную вероятность того, что первые два наблюдения дали $j$ и $k$, если известно, что $\textbf{X} = r$.
- в) Найти $\textbf{E}(\textbf{X}^2)$ и затем асимптотическое выражение для $\mathrm{Var}(\textbf{X})$ при $N \rightarrow \infty$ (при фиксированном $n$).

#### <mark>Решение.</mark>

Пусть в примере д) $\textbf{Y}$ - наименьший номер, полученный при $n$ извлечениях, если производится случайный выбор с возвращением. А $\textbf{Z}_1, \textbf{Z}_2,\dots,\textbf{Z}_n$ - номера шаров при $n$ извлечениях. $\{\textbf{Z}_i\}$ можно считать взаимно независимыми случайными величинами с равномерным распределением $\textbf{P}\{\textbf{Z}_i=k\}=1/N$ для $i=1,2,\dots,n$ и $k=1,2,\dots,N$.<br>
Найдем совместное распределение $\textbf{P}\{\textbf{X}=r,\ \textbf{Y}=s\}$<br>
Очевидно, что при $s>r$ имеем $$\textbf{P}\{\textbf{X} = r,\ \textbf{Y} = s\}=0$$<br>
Если же $s \le r$, то:<br>
$$\textbf{P}\{\textbf{X} \le r,\ \textbf{Y} \ge s\} = \textbf{P}\{s \le \textbf{Z}_1 \le r,\ s \le \textbf{Z}_2 \le r,\dots,\ s \le \textbf{Z}_n \le r\} = \prod\limits_{i=1}^n\textbf{P}\{s \le \textbf{Z}_i \le r\} = \prod\limits_{i=1}^n\sum\limits_{j=s}^r\textbf{P}\{\textbf{Z}_i=j\}=\left(\frac{r-s+1}{N}\right)^n$$<br>
Поэтому при $s=r$ получим $$\textbf{P}\{\textbf{X} = r,\ \textbf{Y} = s\} = 1/N^n$$<br>
А в случае $r > s$ имеем:<br>
$$\textbf{P}\{\textbf{X} \le r,\ \textbf{Y} \ge s\} = \textbf{P}\{\textbf{X} = r,\ \textbf{Y} =s\} + \textbf{P}\{\textbf{X} \le r-1,\ \textbf{Y} \ge s\} + \textbf{P}\{\textbf{X} \le r,\ \textbf{Y} \ge s + 1\} - \textbf{P}\{\textbf{X} \le r-1,\ \textbf{Y} \ge s+1\}$$<br>
т.е. 
$$\textbf{P}\{\textbf{X}=r,\ \textbf{Y}=s\} = \frac{(r-s+1)^n-2(r-s)^n+(r-s-1)^n}{N^n}$$<br>
Поэтому ответ на вопрос а)
$$\begin{matrix}
    \textbf{P}\{\textbf{X}=r,\ \textbf{Y}=s\} & =
    & \left\{
    \begin{matrix}
    0 & \mbox{если } s > r \\
    1/N^n & \mbox{если } s = r \\
    \frac{(r-s+1)^n-2(r-s)^n+(r-s-1)^n}{N^n} & \mbox{если } s <  r \\
    \end{matrix} \right.
\end{matrix}$$
Как видно при $n=2$ совместное распределение имеет вид $1/N^2$ при $s \le r$ и $0$ при $s > r$.

Рассмотрим теперь задачу б). Найдем искомую условную вероятность: $$\textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k\ |\ \textbf{X}=r\} = \frac{\textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k,\ \textbf{X}=r\}}{\textbf{P}\{\textbf{X}=r\}}$$ Как уже устновлено знаменатель $$\textbf{P}\{\textbf{X}=r\}=\frac{r^n-(r-1)^n}{N^n}$$ Вычисление числителя зависит от нескольких случаев. Рассмотрим их.
- Случай $j > r$ или $k > r$. В этом случае, очевидно, числитель и искомая условная вероятность равны $0$
- Случай $j < r$ и $k < r$. В этом случае $$\textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k,\ \textbf{X} \le r\} = \textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k,\ \textbf{Z}_3 \le r,\ \dots,\  \textbf{Z}_n \le r\} = \frac{1}{N^2}\prod\limits_{i=3}^n\sum\limits_{t=1}^r\textbf{P}\{\textbf{X}_i=t\} = \frac{r^{n-2}}{N^n}$$ поэтому $$\textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k,\ \textbf{X} = r\} = \textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k,\ \textbf{X} \le r\} - \textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k,\ \textbf{X} \le r-1\} = \frac{r^{n-2}-(r-1)^{n-2}}{N^n}$$
- Случай $j = r$ и $k < r$, либо $k = r$ и $j < r$. В этом случае $$\textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k,\ \textbf{X} = r\} =  \textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k,\ \textbf{Z}_3 \le r,\ \dots,\  \textbf{Z}_n \le r\} = \frac{r^{n-2}}{N^n}$$

Таким образом ответ на вопрос б):
$$\begin{matrix}
    \textbf{P}\{\textbf{Z}_1=j,\ \textbf{Z}_2=k\ |\ \textbf{X} = r\} & =
    & \left\{
    \begin{matrix}
    0 & \mbox{если } j > r \mbox{ или } k > r \\
    \frac{r^{n-2}-(r-1)^{n-2}}{r^n-(r-1)^n} & \mbox{если } j < r \mbox{ и } k < r \\
    \frac{r^{n-2}}{r^n-(r-1)^n} & \mbox{если } j = r \mbox{ и } k < r, \mbox{ либо } k = r \mbox{ и } j < r \\
    \end{matrix} \right.
\end{matrix}$$

Рассмотрим задачу в). По определению $$\textbf{E}(\textbf{X}^2)=\sum\limits_{k=1}^Nk^2p_k=\sum\limits_{k=1}^Nk^2\frac{k^n-(k-1)^n}{N^n}=N^{-n}\sum\limits_{k=1}^N(k^{n+2}-k^2(k-1)^n)$$ Заметим, что $$k^2(k-1)^n = (k-1)^{n+2} + (2k-1)(k-1)^n = (k-1)^{n+2}+2(k-1)^{n+1}+(k-1)^n$$<br> Поэтому сумму в последнем выражении можно переписать как $$\sum\limits_{k=1}^N(k^{n+2}-(k-1)^{n+2}) - 2\sum\limits_{k=1}^N(k-1)^{n+1}-\sum\limits_{k=1}^N(k-1)^n$$ Здесь первая сумма равна $N^{n+2}$, а две другие при больших $N$ можно оценить интегралами $$-2\int\limits_0^Nx^{n+1}dx-\int\limits_0^Nx^ndx = -\frac{2N^{n+2}}{n+2}-\frac{N^{n+1}}{n+1}$$<br>
Таким образом $$\textbf{E}(\textbf{X}^2)\approx N^{-n}\left(N^{n+2}-\frac{2N^{n+2}}{n+2}-\frac{N^{n+1}}{n+1}\right)=N^2-\frac{2N^2}{n+2}-\frac{N}{n+1}$$
При больших $N$ последним слагаемым можно принебречь по сравнению с членами  $N^2$. Поэтому $$\textbf{E}(\textbf{X}^2)\approx \frac{N^2n}{n+2}$$ Оценка $$\textbf{E}(\textbf{X})\approx \frac{Nn}{n+1}$$ была получена в примере д). Таким образом ответ на вопрос в): $$\mathrm{Var}(\textbf{X})=\textbf{E}(\textbf{X}^2)-\textbf{E}(\textbf{X})^2\approx\frac{N^2n}{n+2}-\frac{N^2n^2}{(n+1)^2}=\frac{N^2n}{(n+2)(n+1)^2}$$

## Задача № 10 🪙

### <mark>Условие.</mark>

*Имитация симметричной монеты.* Для данной несимметричной монеты, такой, что вероятность выпадения герба равна $\alpha$, имитируем правильную монету следующим образом. Бросаем несимметричную монету дважды. Считаем комбинацию ГР за успех, а РГ - за неудачу; если ни одно из этих событий не произошло, то повторяем бросания до осуществления любого из них. a) Показать, что эта модель приводит к испытаниям Бернулли с $p=1/2$. б) Найти распределение и математическое ожидание числа бросаний до появления одного из событий ГР или РГ.

#### <mark>Решение.</mark>

Пусть $p=\alpha$ и $q=1-\alpha$. Найдем структуру пространства элементарных событий $\mathfrak{S}$, выписывая в таблицу элементарные события и их вероятности:<br>

|#|событие (серия)| вероятность<br> события|исход|
|---|---|---|---|
|2.1|ГР|$pq$|Успех|
|2.2|РГ|$pq$|Неудача|
|4.1|ГГГР|$p^3q$|Успех|
|4.2|ГГРГ|$p^3q$|Неудача|
|4.3|РРГР|$pq^3$|Успех|
|4.4|РРРГ|$pq^3$|Неудача|
|6.1|ГГГГГР|$p^5q$|Успех|
|6.2|ГГГГРГ|$p^5q$|Неудача|
|6.3|ГГРРГР|$p^3q^3$|Успех|
|6.4|ГГРРРГ|$p^3q^3$|Неудача|
|6.5|РРГГГР|$p^3q^3$|Успех|
|6.6|РРГГРГ|$p^3q^3$|Неудача|
|6.7|РРРРГР|$pq^5$|Успех|
|6.8|РРРРРГ|$pq^5$|Неудача|
|...|...|...|...|

Как видно вероятность успеха $$\textbf{P}\{\mbox{Испытание закончилось успехом}\} = pq + (p^3q+pq^3)+(p^5q+2p^3q^3+pq^5)+(p^7q+3p^5q^3+3p^3q^5+pq^7)+\dots =$$$$= pq(1+(p^2+q^2)+(p^2+q^2)^2+(p^2+q^2)^3+\dots)$$
Очевидно, что имеется $k$ успешных серий длинной $2k$. Покажем, что вероятность того, что серия длиной $2k$ закончится успехом будет
$$pq(p^2+q^2)^{k-1}$$
Будем считать удачной парой подбрасываний, если она окончилась результатом ГР и неудачной парой, если она окончилась результатами ГГ или РР. Тогда успешная серия гербов и решек длины $2k$ равносильна последовательности независимых пар подбрасываний длины $k$ вида ННН...У, где последовательность из $k-1$ букв Н (одна из двух неудачных пар) заканчивается буквой У (удачной парой ГР).  Вероятность каждой из пар ГР, РР, ГГ, очевидно, равна $pq$, $q^2$ и $p^2$ соответственно. Поэтому $\textbf{P}\{\mbox{Н}\} = p^2+q^2$ и $\textbf{P}\{\mbox{У}\} = pq$. И т.к. пары испытаний независимы, то
$$\textbf{P}\{\underbrace{\mbox{ННН}\dots\mbox{У}}_{k}\}=pq(p^2+q^2)^{k-1}$$
Таким образом получаем сумму геом. прогрессии:
$$\textbf{P}\{\mbox{Испытание закончилось успехом}\}=pq\sum\limits_{k=1}^{\infty}(p^2+q^2)^{k-1}=\frac{pq}{1-p^2-q^2}$$
Т.к. $p+q=1$, то $1-p^2-q^2=2pq$ и следовательно вероятность успеха равна $1/2$. Таким образом получили модель Бернулли с $p=1/2$.<br>
Решим задачу б). Пусть $\textbf{X}$ число бросаний до появления либо ГР, либо РГ (одним бросанием считаем пару подбрасываний монеты). Удачная пара подбрасываний тогда будет либо ГР, либо РГ, а неудачная будет либо ГГ, либо РР. Тогда $\textbf{P}\{\mbox{Н}\} = p^2+q^2$ и $\textbf{P}\{\mbox{У}\} = 2pq$ и поэтому распределение будет
$$\textbf{P}\{\textbf{X}=k\} = \textbf{P}\{\underbrace{\mbox{ННН}\dots\mbox{У}}_k\}=2pq(p^2+q^2)^{k-1}$$
Найдем мат. ожидание  $\textbf{X}$
$$\textbf{E}(\textbf{X})=\sum\limits_{k=1}^\infty k\cdot 2pq(p^2+q^2)^{k-1}$$
Пусть $t = p^2+q^2$, тогда
$$\textbf{E}(\textbf{X})=(1-t)\sum\limits_{k=1}^{\infty}kt^{k-1}$$
Последняя сумма - это производная по $t$ суммы геом. прогрессии $\left(\sum\limits_{k=1}^\infty t^k\right)^{'}=\frac{1}{(1-t)^2}$. Таким образом
$$\textbf{E}(\textbf{X}) = \frac{1}{1-t} = \frac{1}{2pq} = \frac{1}{2\alpha(1-\alpha)}$$

# **Глава XI.<br/>Целочисленные случайные величины. Производящие функции.**

## Задача № 1 🧮

### <mark>Условие.</mark>

Пусть $\textbf{X}$ - случайная величина с производящей функцией $P(s)$. Найти производящие функции случайных величин $\textbf{X}+1$ и $2\textbf{X}$.

#### <mark>Решение.</mark>

Пусть $p_k = \textbf{P}\{\textbf{X}=k\}$ при $k=0,1,2,\dots$. Тогда $$P(s)=\sum\limits_{k=0}^{\infty}p_k s^k$$
Найдем производящую функцию случайной величины $\textbf{Y}=\textbf{X}+1$<br>
Пусть $q_k = \textbf{P}\{\textbf{Y}=k\}$ при $k=0,1,2,\dots$. Тогда $$q_k = \textbf{P}\{\textbf{X}=k-1\}$$ Т.е.
$$\begin{matrix}
    q_k & =
    & \left\{
    \begin{matrix}
    0 & \mbox{если } k=0 \\
    p_{k-1} & \mbox{если } k = 1,2,3,\dots \\
    \end{matrix} \right.
\end{matrix}$$
Поэтому производящая функция $\textbf{Y}$ будет иметь вид $$Q(s) = \sum\limits_{k=0}^{\infty}q_k s^k = \sum\limits_{k=1}^{\infty}p_{k-1} s^k = s\sum\limits_{k=1}^{\infty}p_{k-1} s^{k-1} = sP(s) $$
<br>
Найдем производящую функцию случайной величины $\textbf{Z}=2\textbf{X}$<br>
Пусть $r_k = \textbf{P}\{\textbf{Z}=k\}$ при $k=0,1,2,\dots$. Тогда $$r_k = \textbf{P}\{2\textbf{X}=k\} = \textbf{P}\{\textbf{X}=k/2\}$$
Т.е.
$$\begin{matrix}
    r_k & =
    & \left\{
    \begin{matrix}
    0 & \mbox{если } k \mbox{ нечетно} \\
    p_{k/2} & \mbox{если } k \mbox{ четно}\\
    \end{matrix} \right.
\end{matrix}$$
Поэтому производящая функция $\textbf{Z}$ будет иметь вид $$R(s) = \sum\limits_{k=0}^{\infty}r_k s^k = \sum\limits_{k=0}^{\infty}r_{2k} s^{2k} = \sum\limits_{k=0}^{\infty}p_{k} s^{2k} = P(s^2) $$

## Задача № 2 🧮

### <mark>Условие.</mark>

Найти производящие функции следующих последовательностей 
* а) $\textbf{P}\{\textbf{X} \le n\}$;
* б) $\textbf{P}\{\textbf{X} \lt n\}$;
* в) $\textbf{P}\{\textbf{X} \ge n\}$; 
* г) $\textbf{P}\{\textbf{X} \gt n+1\}$; 
* д) $\textbf{P}\{\textbf{X} = 2n\}$;

#### <mark>Решение.</mark>

Исходим из условия задачи 1: Если $\textbf{P}\{\textbf{X} = k\} = p_k$, то $P(s)=\sum\limits_{k=0}^{\infty}p_k s^k$; 
* a) Очевидно, что  $\textbf{P}\{\textbf{X} \le n\} = \textbf{P}\{\bigcup\limits_{k=0}^{n}\{\textbf{X}=k\}\} = \sum\limits_{k=0}^{n}p_k = q_n$. Требуется найти производящую функцию $Q(s)=\sum\limits_{n=0}^{\infty}q_n s^n$. Рассмотрим последовательность $a_n=1$, где $n=0,1,2,\dots$. Очевидно последовательность $\{q_n\}$ в точности равна свертке $\{p_n\}$ и $\{a_n\}$, т.е. $\{q_n\}=\{p_n\}*\{a_n\}$, поэтому $Q(s)=P(s)A(s)$, где производящая функция последовательности $\{a_n\}$ есть $A(s)=\sum\limits_{n=0}^{\infty}s^n=1/(1-s)$. Таким образом
$$Q(s)=\frac{P(s)}{1-s}$$
* б) Очевидно, что $\textbf{P}\{\textbf{X} \lt n\} = \sum\limits_{k=0}^{n-1}p_k = q_n$ Рассмотрим последовательность $\{a_n\}=\{0,1,1,\dots\}$ Очевидно последовательность $\{q_n\}=\{p_n\}*\{a_n\}$, поэтому искомая производящая функция $Q(s)=P(s)A(s)$, где производящая функция последовательности $\{a_n\}$ есть $A(s)=\sum\limits_{n=1}^{\infty}s^n=s/(1-s)$. Таким образом
$$Q(s)=\frac{sP(s)}{1-s}$$
* в) Пусть $q_n=\textbf{P}\{\textbf{X} \lt n\}$ из задачи б) и $r_n = \textbf{P}\{\textbf{X} \ge n\} = 1 - \textbf{P}\{\textbf{X} \lt n\} = 1 - q_n$, тогда искомая производящая функция
$$R(s) = \sum\limits_{n=0}^{\infty}r_n s^n = \sum\limits_{n=0}^{\infty}(1-q_n) s^n = \sum\limits_{n=0}^{\infty}s^n - \sum\limits_{n=0}^{\infty}q_n s^n = \frac{1}{1-s}-\frac{sP(s)}{1-s}=\frac{1-sP(s)}{1-s}$$
* г) Пусть $r_n=\textbf{P}\{\textbf{X} \ge n\}$ из задачи в) и $q_n=\textbf{P}\{\textbf{X} \gt n+1\} = \textbf{P}\{\textbf{X} \ge n+2\} = r_{n+2}$<br>
Тогда искомая производящая функция $$Q(s)=\sum\limits_{n=0}^{\infty}q_n s^n = \sum\limits_{n=0}^{\infty}r_{n+2} s^n$$
или 
$$s^2Q(s) = \sum\limits_{n=0}^{\infty}r_{n+2} s^{n+2} = R(s) - r_0 - r_1 s$$
где очевидно $r_0 = 1$ и $r_1 = 1 - \textbf{P}\{\textbf{X}=0\} = 1 - p_0 = 1 - P(0)$.<br>
Поэтому 
$$s^2 Q(s) = \frac{1-sP(s)}{1-s} - 1 - (1-p_0)s = \frac{s(s-P(s))}{1-s} + p_0 s $$
Таким образом искомая функция:
$$Q(s) = \frac{1-s^{-1}P(s)}{1-s} + p_0 s^{-1} $$
* д) Пусть $q_n = \textbf{P}\{\textbf{X}=2n\} = p_{2n}$ Тогда искомая производящая функция
$$Q(s) = \sum\limits_{n=0}^{\infty}q_n s^n = \sum\limits_{n=0}^{\infty}p_{2n} s^n$$
Заметим, что
$$P(s^{1/2}) = Q(s) + \sum\limits_{n=1}^{\infty}p_{2n-1} s^{(2n-1)/2} = Q(s) + R(s)$$
и, что
$$P(-s^{1/2}) = Q(s) - R(s)$$
Поэтому искомая функция:
$$Q(s) = \frac{P(\sqrt{s})+P(-\sqrt{s})}{2}$$

## Задача № 3 🧮

### <mark>Условие.</mark>

Рассмотрим последовательности испытаний Бернулли. Обозначим через $u_n$ вероятность того, что комбинация УН впервые появится при $(n-1)$-м и $n$-м испытаниях. Найти производящую функцию, среднее значение и дисперсию. 

#### <mark>Решение.</mark>

Во-первых, очевидно, что $u_0=u_1=0$.<br>
Во-вторых, если $n>1$, то целевое событие задачи (вероятность которого $u_n$) распадается на $2$ несовместных подсобытия:
* 1) первое испытание закончилось неудачей и тогда вероятность целевого события будет $\textbf{P}\{Н\}\cdot u_{n-1}=qu_{n-1}$ 
* 2) первое испытание закончилось удачей и тогда вероятность целевого события будет: $$\textbf{P}\{\underbrace{\mbox{УУУ}\dots\mbox{Н}}_{n}\}=p^{n-1}q$$

Таким образом полная вероятность будет
$$u_n = qu_{n-1}+p^{n-1}q$$
Отсюда получаем выражение для искомой производящей функции
$$U(s)=\sum\limits_{n=0}^{\infty}u_n s^n=\sum\limits_{n=2}^{\infty}(qu_{n-1}+p^{n-1}q) s^n$$
или 
$$U(s) = qs\sum\limits_{n=1}^{\infty}u_n s^n + qs\sum\limits_{n=1}^{\infty}(ps)^n = qsU(s) + \frac{pqs^2}{1-ps}$$
Таким образом
$$U(s)=\frac{pqs^2}{(1-ps)(1-qs)}$$
Отсюда производная производящей функции будет
$$U^{\prime}(s) = \frac{2pqs(1-ps)(1-qs)+pqs^2(p(1-qs)+q(1-ps))}{(1-qs)^2(1-ps)^2} = \frac{pqs(2-s)}{(1-ps)^2(1-qs)^2}$$
Поэтому мат. ожидание сл. величины $\textbf{X}$ с распределением вероятностей $u_n$ будет
$$\textbf{E}(\textbf{X})=U^{\prime}(1) = \frac{pq}{p^2q^2}=\frac{1}{pq}$$
Для вычисления дисперсии $\textbf{X}$ необходимо найти $U^{\prime\prime}(1)
$. Пусть $U^\prime(s)=\frac{pqs(2-s)}{(1-ps)^2(1-qs)^2} = \frac{f(s)}{g(s)}$, т.е. $f(s)=pqs(2-s),\ g(s)=(1-ps)^2(1-qs)^2$, тогда $U^{\prime\prime}(1)=\frac{f^\prime(1)g(1)-g^\prime(1)f(1)}{g^2(1)}$, где 
$$f(1)=pq$$
$$f'(1)=2pq(1-s)|_{s=1}=0$$
$$g(1)=p^2q^2$$
$$g^\prime(1)=((-2p+2p^2s)(1-qs)^2+(-2q-2q^2s)(1-ps)^2)|_{s=1}=-2pq(p^2+q^2)$$
Таким образом
$$U^{\prime\prime}(1)=\frac{2pq(p^2+q^2)pq}{p^4q^4}=\frac{2(1-2pq)}{p^2q^2}=\frac{2-4pq}{p^2q^2}$$
Поэтому
$$\mathrm{Var}(\textbf{X})=U^{\prime\prime}(1)+U^{\prime}(1)-(U^{\prime}(1))^2 = \frac{1-4pq}{p^2q^2}+\frac{1}{pq}=\frac{1-3pq}{p^2q^2}$$


## Задача № 5 🧮

### <mark>Условие.</mark>

Пусть $a_n$ - число способов, которыми можно получить сумму в $n$ очков при произвольном числе бросаний правильной кости. Показать, что производящая функция последовательности $\{a_n\}$ равна $(1-s-s^2-s^3-s^4-s^5-s^6)^{-1}-1$

#### <mark>Решение.</mark>

Очевидно $a_0=0, a_1=1$<br>
Пусть $b_n^k$ будет число способов получить $n$ очков при $k$ бросаниях правильной кости. Тогда $a_n=\sum\limits_{k=1}^{\infty}b_n^k$ (для любого фиксированного $n$ эта сумма конечна, т.к. $b_n^k = 0$ при любом $k>n$).<br> 
Если $k=1$, то $b_n^1=1$ только при $n=1,2,3,4,5,6$ и $0$ во всех остальных случаях.<br>
Eсли $k>1$, то при $n>6$ имеем
$$b_n^k=\sum\limits_{i=1}^{6}b_{n-i}^{k-1}\ \ \ (*)$$
(Потому что при первом бросании возможны $6$ различных исходов, каждый из которых в следующих $k-1$ бросаниях дает $b_{n-i}^{k-1}$ способов получить общую сумму $n$ очков, где $i=1,2,3,4,5,6$ это результат одного из $6$ возможных исходов первого бросания).<br>
Поэтому
$$a_n=\sum\limits_{i=1}^{6}\sum\limits_{k=1}^{\infty}b_{n-i}^{k-1}=a_{n-1}+a_{n-2}+a_{n-3}+a_{n-4}+a_{n-5}+a_{n-6}$$
(здесь $b_{n-i}^0$ считаем нулями)<br>
Найдем $a_i$ при $i=2,3,4,5,6$ (тут пользуемся также тем, что $b_k^k=1$ для всех $k>0$):
* $a_2=b_2^1+b_2^2=1 + 1$
* $a_3=b_3^1+b_3^2+b_3^3=1 + 2 + 1 = 4$<br>
($b_3^2=b_2^2+b_1^1=1+1=2$)
* $a_4=b_4^1+b_4^2+b_4^3+b_4^4=1 + 3 + 3 + 1=8$<br>
($b_4^2=b_3^1+b_2^1+b_1^1=1+1+1=3,\ b_4^3=b_2^2+b_3^2=1+2=3$)
* $a_5 = 1 + 4 + 6 + 4 + 1 =16$ (аналогичные рекуррентные формулы)
* $a_6 = 1 + 5 + 10 + 10 +5 +1 = 32$ (аналогичные рекуррентные формулы)

Пользуясь $(*)$ построим для наглядности таблицу $b_n^k$ и вычислим первые $10$ значений $a_n$ (суммы ненулевых $b_n^k$ по первым $10$ столбцам).

| $k$ \ $n$ | 1 | 2 | 3 | 4| 5 | 6 | 7 | 8 | 9 | 10 |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 1 |1|1|1|1|1|1|0|0|0|0|
| 2 |0|1|2|3|4|5|6|5|4|3|
| 3 |0|0|1|3|6|10|15|21|25|27|
| 4 |0|0|0|1|4|10|20|35|56|80|
| 5 |0|0|0|0|1|5|15|35|70|126|
| 6 |0|0|0|0|0|1|6|21|56|126|
| 7 |0|0|0|0|0|0|1|7|28|84|
| 8 |0|0|0|0|0|0|0|1|8|36|
| 9 |0|0|0|0|0|0|0|0|1|9|
| 10|0|0|0|0|0|0|0|0|0|1|
|$a_n$|1|2|4|8|16|32|63|125|248|492|

Итак, мы имеем $a_0=0, a_n=2^{n-1}$ при $n=1,2,3,4,5,6$ и если $n>6$, то
$$a_n=a_{n-1}+a_{n-2}+a_{n-3}+a_{n-4}+a_{n-5}+a_{n-6}$$
Т.е. для искомой производящей функции справедливо следующее:
$$A(s)=\sum\limits_{n=0}^{\infty}a_n s^n=\sum\limits_{n=1}^{6}2^{n-1} s^n+ \sum_{n=7}^{\infty}s^n\sum\limits_{k=1}^{6}a_{n-k}$$
или
$$A(s)=\sum\limits_{n=1}^{6}2^{n-1} s^n + \sum\limits_{k=1}^6 s^i\sum\limits_{n=7-k}^{\infty}a_n s^n$$
или
$$A(s)=\sum\limits_{n=1}^{6}2^{n-1} s^n + \sum\limits_{k=1}^{6}s^k\left(A(s)-\sum\limits_{j=1}^{6-k}2^{j-1}s^j\right)$$
группируя все множители при $A(s)$ слева получаем:
$$A(s)(1-s-s^2-s^3-s^4-s^5-s^6)=\sum\limits_{n=1}^{6}2^{n-1} s^n - \sum\limits_{k=1}^{5}s^k\left(\sum\limits_{j=1}^{6-k}2^{j-1}s^j\right)$$
или
$$A(s)(1-s-s^2-s^3-s^4-s^5-s^6)=\sum\limits_{n=1}^{6}s^n\left(2^{n-1}-\sum\limits_{j=1}^{n-1}2^{j-1}\right)=\sum\limits_{n=1}^{6}s^n$$
Таким образом
$$A(s)=\frac{1}{1-s-s^2-s^3-s^4-s^5-s^6}-1$$
что и требовалось показать.



## Задача № 6 🧮

### <mark>Условие.</mark>

Пусть $a_n$ - число способов, которыми $n-2$ (непересекающихся) диагоналей разбивают выпуклый (n+1)-угольник $P_0 P_1\dots P_n$ на треугольники$^{1)}$. Положим $a_1=1$. Показать, что при $n \ge 2$
$$a_n=a_1 a_{n-1} + a_2 a_{n-2}+\dots +a_{n-1}a_1$$
Найти производящую функцию и явные выражения для $a_n$.<br>
*Указание.* Предположить, что одна из диагоналей проходит через $P_0$ и принять за $k$ наименьший индекс, такой, что $P_0 P_k$ является диагональю.
<br>
<br>
<br>
$^{1)}$ Эта задача обсуждается в книге Polya G. Mathematics and plausible reasoning, Princeton University Press, 1954, p. 102 [Имеется перевод: Пойа Д., Математика и правдоподобные рассуждения. - М.: ИЛ, 1957; М.: Наука, 1975, с. 123.]

#### <mark>Решение.</mark>

Пускай, согласно указанию, $b_k$ число способов триангуляции $(n+1)$-угольника, в которых существует диагональ проходящая через $P_0$, а $k$ наименьший индекс, такой, что $P_0 P_k$ является диагональю. Понятно, что $k$ пробегает $n-2$ значения от $2$ до $n-1$. Тогда
$$a_n=c_n + \sum\limits_{k=2}^{n-1}b_k$$
где $c_n$ это число способов триангуляции $(n+1)$-угольника, в которых через $P_0$ диагональ не проходит. Очевидно таких способов будет $$c_n = a_{n-1} = a_1a_{n-1}$$
(в этих способах существует диагональ $P_1P_n$, отсекающая $n$-угольник, который можно триангулиоровать $a_{n-1}$ способами)<br>
Найдем $b_k$.
* 1. $k=2$, тогда диагональ $P_0P_2$ разбивает исходный многоугольник на треугольник $P_0P_1P_2$ и $n$-угольник $P_0P_2\dots P_n$, который можно триангулировать $a_{n-1}$ способами, т.е.
$$b_2=a_{n-1} = a_{n-1}a_1$$
* 2. $n > k > 2$, тогда диагональ $P_0P_k$ разбивает исходный $(n+1)$-угольник на две части $P_0P_kP_{k+1}\dots P_n$ и $P_0P_1\dots P_k$. Первую часть можно триангулировать $a_{n-k+1}$ способом (т.к. это $(n-k+2)$-угольник), а во второй части обязательно должна присутствовать диагональ $P_1P_k$, т.к. в противном случае нарушается условие на минимальность индекса $k$, поэтому кол-во способов триангуляций второй части равно числу способов триангуляций $k$-угольника $P_1P_2\dots P_k$, т.е. $a_{k-1}$. Таким образом
$$b_k = a_{n-k+1}a_{k-1}$$
В конечном итоге получаем:
$$a_n = a_1a_{n-1} + \sum\limits_{k=3}^{n-1}a_{k-1}a_{n-k+1}+a_{n-1}a_1$$
или
$$a_n=a_1 a_{n-1} + a_2 a_{n-2}+\dots +a_{n-1}a_1$$
что и требовалось показать.<br>

Пусть $a_0=0$. Тогда согласно найденному выше рекурентному соотношению свертка $\{c_n\}=\{a_n\}*\{a_n\}$ отличается от последовательности $\{a_n\}$ только при $n=1$: $c_1=0 \neq a_1=1$, поэтому искомая производящая функция удовлетворяет равенству
$$A^2(s)+s=A(s)$$
т.е. 
$$A(s)=\frac{1\pm\sqrt{1-4s}}{2}$$
Т.к. $a_n \ge 0$, то
$$A(s)=\frac{1 - \sqrt{1-4s}}{2}$$
(в противном случае $A(s)=1-s-s^2 + O(s^3)$ в окрестности $0$, т.е.,например, $a_1<0$, что недопустимо). 
Найдем явный вид $a_n$. Как известно для любого $a$ и любого $-1<t<1$ справедлива формула бинома Ньютона
$$(1+t)^a = 1 + {a \choose 1}t + {a \choose 2}t^2+{a \choose 3}t^3+\dots$$
Поэтому
$$A(s) = \frac{1 - \sqrt{1-4s}}{2} = \frac{1}{2} - \frac{1}{2}\left(\sum\limits_{n=0}^{\infty}{0.5 \choose n}(-4)^n s^n\right)=\frac{1}{2}\sum\limits_{n=1}^{\infty}{0.5 \choose n}(-1)^{n+1}4^n s^n$$
т.е.
$$a_n=\frac{1}{2}{0.5 \choose n}(-1)^{n+1}4^n$$
Согласно задаче $4$ $\S 12$ главы II
$$\frac{1}{n}{2n-2 \choose n-1}2^{-2n+1}=(-1)^{n+1}{0.5 \choose n}$$
поэтому получаем
$$a_n=\frac{1}{n}{2n-2 \choose n-1}$$